{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Notes from meeting\n",
    "- Start with simpler encoding\n",
    "- Don't use > 1bp SNPs right now\n",
    "- Add count for mutation IDs\n",
    "- Get set of chromosomes for each class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from os import listdir\n",
    "from statistics import mean\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['OV-US',\n",
       " 'organize_data.py',\n",
       " 'COAD-US',\n",
       " '.DS_Store',\n",
       " 'list_data_dirs.py',\n",
       " 'UCEC-US',\n",
       " 'PRAD-US',\n",
       " 'KIRC-US',\n",
       " 'SKCM-US',\n",
       " 'THCA-US',\n",
       " 'LGG-US',\n",
       " 'GBM-US',\n",
       " 'LUSC-US',\n",
       " 'BLCA-US',\n",
       " 'BRCA-US',\n",
       " '.remove_duplicates.py.swo']"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "listdir('data')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Gets unique mutation ids with number of occurrences\n",
    "# Returns new dataframe with cancer type, donor id, mutation ids and corresponding counts\n",
    "def mutation_counts(df):\n",
    "    cancer_type = df.project_code[0]\n",
    "    donor_id = df.icgc_donor_id[0] + cancer_type\n",
    "    smaller_df = df[['icgc_mutation_id']]\n",
    "    unique_ids, counts = np.unique(smaller_df.to_numpy(), \n",
    "                                   return_counts=True,\n",
    "                                  )\n",
    "    new_df = pd.DataFrame({\"CancerType\": cancer_type,\n",
    "                           \"DonorId\": donor_id,\n",
    "                           \"MutationId\": unique_ids, \n",
    "                           \"Count\": counts})\n",
    "    return new_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The mutation counts for a cancer type is returned\n",
    "def get_mutation_frames(cancer_code):\n",
    "    DATA_PATH = 'data/'\n",
    "    PATH = DATA_PATH + cancer_code + '/'\n",
    "    \n",
    "    all_files = [file for file in listdir(PATH)]\n",
    "    \n",
    "    final_df = None\n",
    "    \n",
    "    for file in all_files:\n",
    "        df = pd.read_csv(PATH + file, sep='\\t')\n",
    "        final_df = pd.concat([final_df, mutation_counts(df)])\n",
    "        \n",
    "    return final_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/nima/miniconda3/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3254: DtypeWarning: Columns (8) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  if (await self.run_code(code, result,  async_=asy)):\n"
     ]
    }
   ],
   "source": [
    "# Go through all cancer types and get mutation dataframes\n",
    "# Concatenate all dataframes\n",
    "# This results in one dataframe for all cancer types\n",
    "\n",
    "folders = [file for file in listdir('data') if 'US' in file]\n",
    "all_cancer_df = None\n",
    "for cancer in folders:\n",
    "    all_cancer_df = pd.concat([all_cancer_df, \n",
    "                               get_mutation_frames(cancer)])\n",
    "    \n",
    "all_cancer_df.reset_index(drop=True, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CancerType</th>\n",
       "      <th>DonorId</th>\n",
       "      <th>MutationId</th>\n",
       "      <th>Count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OV-US</td>\n",
       "      <td>DO28611OV-US</td>\n",
       "      <td>MU129325554</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OV-US</td>\n",
       "      <td>DO28611OV-US</td>\n",
       "      <td>MU130100811</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OV-US</td>\n",
       "      <td>DO28611OV-US</td>\n",
       "      <td>MU130473495</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OV-US</td>\n",
       "      <td>DO28611OV-US</td>\n",
       "      <td>MU130475900</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OV-US</td>\n",
       "      <td>DO28611OV-US</td>\n",
       "      <td>MU130477835</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2534368</th>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>DO5857BRCA-US</td>\n",
       "      <td>MU5271753</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2534369</th>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>DO5857BRCA-US</td>\n",
       "      <td>MU5272943</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2534370</th>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>DO5857BRCA-US</td>\n",
       "      <td>MU5272953</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2534371</th>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>DO5857BRCA-US</td>\n",
       "      <td>MU5468694</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2534372</th>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>DO5857BRCA-US</td>\n",
       "      <td>MU5468709</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2534373 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        CancerType        DonorId   MutationId  Count\n",
       "0            OV-US   DO28611OV-US  MU129325554      4\n",
       "1            OV-US   DO28611OV-US  MU130100811     29\n",
       "2            OV-US   DO28611OV-US  MU130473495      3\n",
       "3            OV-US   DO28611OV-US  MU130475900     17\n",
       "4            OV-US   DO28611OV-US  MU130477835      8\n",
       "...            ...            ...          ...    ...\n",
       "2534368    BRCA-US  DO5857BRCA-US    MU5271753     16\n",
       "2534369    BRCA-US  DO5857BRCA-US    MU5272943      7\n",
       "2534370    BRCA-US  DO5857BRCA-US    MU5272953      7\n",
       "2534371    BRCA-US  DO5857BRCA-US    MU5468694      7\n",
       "2534372    BRCA-US  DO5857BRCA-US    MU5468709      7\n",
       "\n",
       "[2534373 rows x 4 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_cancer_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_mutations, counts = np.unique(np.array(all_cancer_df.MutationId), return_counts=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "mutation_df = pd.DataFrame( {'MutationID': unique_mutations, \n",
    "                             'Counts': counts} )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "mutation_df_filtered = mutation_df[mutation_df.Counts > 4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/nima/miniconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \"\"\"Entry point for launching an IPython kernel.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MutationID</th>\n",
       "      <th>Counts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1186458</th>\n",
       "      <td>MU1899169</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1236057</th>\n",
       "      <td>MU1957569</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1236120</th>\n",
       "      <td>MU1957631</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1236182</th>\n",
       "      <td>MU1957694</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1236245</th>\n",
       "      <td>MU1957756</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1424496</th>\n",
       "      <td>MU4468</td>\n",
       "      <td>153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1631077</th>\n",
       "      <td>MU866</td>\n",
       "      <td>238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1587418</th>\n",
       "      <td>MU62030</td>\n",
       "      <td>365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1046482</th>\n",
       "      <td>MU131898417</td>\n",
       "      <td>400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1044295</th>\n",
       "      <td>MU131867962</td>\n",
       "      <td>400</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20264 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          MutationID  Counts\n",
       "1186458    MU1899169       5\n",
       "1236057    MU1957569       5\n",
       "1236120    MU1957631       5\n",
       "1236182    MU1957694       5\n",
       "1236245    MU1957756       5\n",
       "...              ...     ...\n",
       "1424496       MU4468     153\n",
       "1631077        MU866     238\n",
       "1587418      MU62030     365\n",
       "1046482  MU131898417     400\n",
       "1044295  MU131867962     400\n",
       "\n",
       "[20264 rows x 2 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutation_df_filtered.sort_values(by=['Counts'], inplace=True)\n",
    "mutation_df_filtered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "mutation_ids = mutation_df_filtered.MutationID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "master_mutations = np.array(mutation_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0., 0., 0., ..., 0., 0., 0.])"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columns = ['DonorIDs', 'CancerType'] + list(master_mutations)\n",
    "df = pd.DataFrame(columns = columns)\n",
    "temp = np.where(master_mutations == 'IS675')\n",
    "np.zeros((len(master_mutations)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "def helper(df):\n",
    "    cancer_type = df.project_code[0]\n",
    "    donor_id = df.icgc_donor_id[0] + cancer_type\n",
    "    smaller_df = df[['icgc_mutation_id']]\n",
    "    unique_ids = np.unique(smaller_df.to_numpy())\n",
    "    return cancer_type, donor_id, unique_ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "def katrinas_function(cancer_code, master_mutations):\n",
    "    \n",
    "    DATA_PATH = 'data/'\n",
    "    PATH = DATA_PATH + cancer_code + '/'\n",
    "    \n",
    "    all_files = [file for file in listdir(PATH)]\n",
    "    columns = ['DonorIDs', 'CancerType'] + list(master_mutations)\n",
    "    final_df = pd.DataFrame(columns = columns)\n",
    "    \n",
    "    for file in all_files:\n",
    "        df = pd.read_csv(PATH + file, sep='\\t')\n",
    "        cancer_type, donor_id, unique_ids = helper(df)\n",
    "        binary = np.zeros((len(master_mutations)))\n",
    "        \n",
    "        for ids in unique_ids:\n",
    "            temp = np.where(master_mutations == ids)\n",
    "            if len(temp) > 0:\n",
    "                binary[temp[0]] = 1\n",
    "        \n",
    "        row = [donor_id, cancer_type] + list(binary)\n",
    "        final_df = pd.concat([final_df, pd.DataFrame([row], columns=columns)], ignore_index=True)\n",
    "    \n",
    "    return final_df\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DonorIDs</th>\n",
       "      <th>CancerType</th>\n",
       "      <th>MU1899169</th>\n",
       "      <th>MU1957569</th>\n",
       "      <th>MU1957631</th>\n",
       "      <th>MU1957694</th>\n",
       "      <th>MU1957756</th>\n",
       "      <th>MU1957895</th>\n",
       "      <th>MU1957974</th>\n",
       "      <th>MU1958009</th>\n",
       "      <th>...</th>\n",
       "      <th>MU130696800</th>\n",
       "      <th>MU122201</th>\n",
       "      <th>MU129795540</th>\n",
       "      <th>MU129540995</th>\n",
       "      <th>MU4885648</th>\n",
       "      <th>MU4468</th>\n",
       "      <th>MU866</th>\n",
       "      <th>MU62030</th>\n",
       "      <th>MU131898417</th>\n",
       "      <th>MU131867962</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DO5944BRCA-US</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DO3067BRCA-US</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DO50036BRCA-US</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DO3275BRCA-US</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DO2978BRCA-US</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1013</th>\n",
       "      <td>DO1697BRCA-US</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1014</th>\n",
       "      <td>DO3793BRCA-US</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1015</th>\n",
       "      <td>DO3400BRCA-US</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1016</th>\n",
       "      <td>DO219593BRCA-US</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1017</th>\n",
       "      <td>DO5857BRCA-US</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1018 rows × 20266 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             DonorIDs CancerType  MU1899169  MU1957569  MU1957631  MU1957694  \\\n",
       "0       DO5944BRCA-US    BRCA-US        0.0        0.0        0.0        0.0   \n",
       "1       DO3067BRCA-US    BRCA-US        0.0        0.0        0.0        0.0   \n",
       "2      DO50036BRCA-US    BRCA-US        0.0        0.0        0.0        0.0   \n",
       "3       DO3275BRCA-US    BRCA-US        0.0        0.0        0.0        0.0   \n",
       "4       DO2978BRCA-US    BRCA-US        0.0        0.0        0.0        0.0   \n",
       "...               ...        ...        ...        ...        ...        ...   \n",
       "1013    DO1697BRCA-US    BRCA-US        0.0        0.0        0.0        0.0   \n",
       "1014    DO3793BRCA-US    BRCA-US        0.0        0.0        0.0        0.0   \n",
       "1015    DO3400BRCA-US    BRCA-US        0.0        0.0        0.0        0.0   \n",
       "1016  DO219593BRCA-US    BRCA-US        0.0        0.0        0.0        0.0   \n",
       "1017    DO5857BRCA-US    BRCA-US        0.0        0.0        0.0        0.0   \n",
       "\n",
       "      MU1957756  MU1957895  MU1957974  MU1958009  ...  MU130696800  MU122201  \\\n",
       "0           0.0        0.0        0.0        0.0  ...          0.0       0.0   \n",
       "1           0.0        0.0        0.0        0.0  ...          0.0       0.0   \n",
       "2           0.0        0.0        0.0        0.0  ...          0.0       0.0   \n",
       "3           0.0        0.0        0.0        0.0  ...          0.0       0.0   \n",
       "4           0.0        0.0        0.0        0.0  ...          0.0       0.0   \n",
       "...         ...        ...        ...        ...  ...          ...       ...   \n",
       "1013        0.0        0.0        0.0        0.0  ...          0.0       0.0   \n",
       "1014        0.0        0.0        0.0        0.0  ...          0.0       0.0   \n",
       "1015        0.0        0.0        0.0        0.0  ...          0.0       0.0   \n",
       "1016        0.0        0.0        0.0        0.0  ...          0.0       0.0   \n",
       "1017        0.0        0.0        0.0        0.0  ...          0.0       0.0   \n",
       "\n",
       "      MU129795540  MU129540995  MU4885648  MU4468  MU866  MU62030  \\\n",
       "0             0.0          0.0        0.0     1.0    0.0      0.0   \n",
       "1             0.0          0.0        0.0     0.0    0.0      0.0   \n",
       "2             0.0          0.0        0.0     0.0    0.0      0.0   \n",
       "3             0.0          0.0        0.0     0.0    0.0      0.0   \n",
       "4             0.0          0.0        0.0     0.0    0.0      0.0   \n",
       "...           ...          ...        ...     ...    ...      ...   \n",
       "1013          0.0          0.0        0.0     0.0    0.0      0.0   \n",
       "1014          0.0          0.0        0.0     0.0    0.0      0.0   \n",
       "1015          0.0          0.0        0.0     0.0    0.0      0.0   \n",
       "1016          0.0          0.0        0.0     0.0    0.0      0.0   \n",
       "1017          0.0          0.0        0.0     0.0    0.0      0.0   \n",
       "\n",
       "      MU131898417  MU131867962  \n",
       "0             0.0          0.0  \n",
       "1             0.0          0.0  \n",
       "2             0.0          0.0  \n",
       "3             0.0          0.0  \n",
       "4             0.0          0.0  \n",
       "...           ...          ...  \n",
       "1013          0.0          0.0  \n",
       "1014          0.0          0.0  \n",
       "1015          0.0          0.0  \n",
       "1016          0.0          0.0  \n",
       "1017          0.0          0.0  \n",
       "\n",
       "[1018 rows x 20266 columns]"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "katrinas_function('BRCA-US', master_mutations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "unique_donors = np.unique(np.array(all_cancer_df.DonorId))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3364"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(unique_donors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_cancer_df_filtered = all_cancer_df[all_cancer_df.Count]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "\"\"\"\n",
    "This function looks at each class and computes the number of donors, \n",
    "average number of mutations and features.\n",
    "\"\"\"\n",
    "def look(cancer_code):\n",
    "    DATA_PATH = 'data/'\n",
    "    PATH = DATA_PATH + cancer_code + '/'\n",
    "    \n",
    "    feat = []\n",
    "    ids = []\n",
    "    donors = []\n",
    "    chromosomes = []\n",
    "    \n",
    "    all_files = [file for file in listdir(PATH)]\n",
    "    \n",
    "    for file in all_files:\n",
    "        df = pd.read_csv(PATH + file, sep='\\t')\n",
    "        feat.append(len(df.columns))\n",
    "        ids.append(len(set(df['icgc_mutation_id'])))\n",
    "        donors.append(len(set(df['icgc_donor_id'])))\n",
    "        \n",
    "        # Repeatedly union the set of chromosomes for each donor\n",
    "        chromosomes += list(df['chromosome'].apply(str))\n",
    "    \n",
    "    print('*' * 40)\n",
    "    print('Project code: ', cancer_code)\n",
    "    \n",
    "    try:\n",
    "        print('Average number of features: {}'.format(mean(feat)))\n",
    "        print('Average number of mutations: {}'.format(mean(ids)))\n",
    "        print('Number of donors: {}'.format(sum(donors)))\n",
    "        print('Chromosomes: {}'.format(set(chromosomes)))\n",
    "    except:\n",
    "        print(NULL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "****************************************\n",
      "Project code:  OV-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 105.53217821782178\n",
      "Number of donors: 404\n",
      "Chromosomes: {'3', '5', '10', '1', '20', '19', '12', '11', '16', '9', '14', '18', '8', '13', '4', '7', '15', 'X', '22', '17', '21', '2', '6'}\n",
      "****************************************\n",
      "Project code:  COAD-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 1949.2\n",
      "Number of donors: 280\n",
      "Chromosomes: {'3', '5', '10', '1', '20', '19', '12', '11', '16', '14', '9', '18', 'Y', '8', '13', '4', '7', '15', 'X', '22', '17', '21', '2', '6'}\n",
      "****************************************\n",
      "Project code:  UCEC-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 3093.343108504399\n",
      "Number of donors: 341\n",
      "Chromosomes: {'3', '5', '10', '1', '20', '19', '12', '11', '16', '14', '9', '18', '8', '13', '4', '7', '15', 'X', '22', '17', '21', '2', '6'}\n",
      "****************************************\n",
      "Project code:  PRAD-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 25.90277777777778\n",
      "Number of donors: 144\n",
      "Chromosomes: {'3', '5', '10', '1', '20', '19', '12', '11', '16', '14', '9', '18', 'Y', '8', '13', '4', '7', '15', 'X', '22', '17', '21', '2', '6'}\n",
      "****************************************\n",
      "Project code:  KIRC-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 106.89270386266094\n",
      "Number of donors: 233\n",
      "Chromosomes: {'3', '5', '10', '1', '20', '19', '12', '11', '16', '9', '14', '18', 'Y', '8', '13', '4', '7', '15', 'X', '22', '17', '21', '2', '6'}\n",
      "****************************************\n",
      "Project code:  SKCM-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 1890.2366666666667\n",
      "Number of donors: 300\n",
      "Chromosomes: {'3', '5', '10', '1', '20', '19', '12', '11', '16', '9', '14', '18', 'Y', '8', '13', '4', '7', '15', 'X', '22', '17', '21', '2', '6'}\n",
      "****************************************\n",
      "Project code:  THCA-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 41.88135593220339\n",
      "Number of donors: 295\n",
      "Chromosomes: {'3', '5', '10', '1', '20', '19', '12', '11', '16', '9', '14', '18', 'Y', '8', '13', '4', '7', '15', 'X', '22', '17', '21', '2', '6'}\n",
      "****************************************\n",
      "Project code:  LGG-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 175.17207792207793\n",
      "Number of donors: 308\n",
      "Chromosomes: {'3', '5', '10', '1', '20', '19', '12', '11', '16', '14', '9', '18', 'Y', '8', '13', '4', '7', '15', 'X', '22', '17', '21', '2', '6'}\n",
      "****************************************\n",
      "Project code:  GBM-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 188.38186813186815\n",
      "Number of donors: 364\n",
      "Chromosomes: {'3', '5', '10', '1', '20', '19', '12', '11', '16', '9', '14', '18', 'Y', '8', '13', '4', '7', '15', 'X', '22', '17', '21', '2', '6'}\n",
      "****************************************\n",
      "Project code:  LUSC-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 9.411157024793388\n",
      "Number of donors: 484\n",
      "Chromosomes: {'3', '5', '10', '1', '20', '19', '12', '11', '9', '14', '8', '13', '7', '15', 'X', '22', '17', '2', '6'}\n",
      "****************************************\n",
      "Project code:  BLCA-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 55.48048780487805\n",
      "Number of donors: 410\n",
      "Chromosomes: {'3', '5', '10', '1', '20', '19', '12', '11', '16', '14', '9', '18', 'Y', '8', '13', '4', '7', '15', 'X', '22', '17', '21', '2', '6'}\n",
      "****************************************\n",
      "Project code:  BRCA-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 130.88801571709234\n",
      "Number of donors: 1018\n",
      "Chromosomes: {'3', '5', '10', '1', '20', '19', '12', '11', '16', '14', '9', '18', 'Y', '8', '13', '4', '7', '15', 'X', '22', '17', '21', '2', '6'}\n"
     ]
    }
   ],
   "source": [
    "folders = [file for file in listdir('data') if 'US' in file]\n",
    "for cancer in folders:\n",
    "    look(cancer)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reading in each file\n",
    "Each file represents one individual. We could probably use the project_code feature as our target/output."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/BRCA-US/simple_somatic_mutation.open-2020-03-02T154602.752.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of features: 42\n",
      "Index(['icgc_mutation_id', 'icgc_donor_id', 'project_code', 'icgc_specimen_id',\n",
      "       'icgc_sample_id', 'matched_icgc_sample_id', 'submitted_sample_id',\n",
      "       'submitted_matched_sample_id', 'chromosome', 'chromosome_start',\n",
      "       'chromosome_end', 'chromosome_strand', 'assembly_version',\n",
      "       'mutation_type', 'reference_genome_allele', 'mutated_from_allele',\n",
      "       'mutated_to_allele', 'quality_score', 'probability', 'total_read_count',\n",
      "       'mutant_allele_read_count', 'verification_status',\n",
      "       'verification_platform', 'biological_validation_status',\n",
      "       'biological_validation_platform', 'consequence_type', 'aa_mutation',\n",
      "       'cds_mutation', 'gene_affected', 'transcript_affected',\n",
      "       'gene_build_version', 'platform', 'experimental_protocol',\n",
      "       'sequencing_strategy', 'base_calling_algorithm', 'alignment_algorithm',\n",
      "       'variation_calling_algorithm', 'other_analysis_algorithm',\n",
      "       'seq_coverage', 'raw_data_repository', 'raw_data_accession',\n",
      "       'initial_data_release_date'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "print('Number of features: {}'.format(len(df.columns)))\n",
    "print(df.columns)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice how for one donor there are duplicate icgc_mutation_id values. Not sure what this means.\n",
    "\n",
    "Number of muations and genes affected do not match :("
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(294,)"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes, counts = np.unique(df.gene_affected.to_numpy(), return_counts=True)\n",
    "genes.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_mutation_id</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_specimen_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>matched_icgc_sample_id</th>\n",
       "      <th>submitted_sample_id</th>\n",
       "      <th>submitted_matched_sample_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>...</th>\n",
       "      <th>experimental_protocol</th>\n",
       "      <th>sequencing_strategy</th>\n",
       "      <th>base_calling_algorithm</th>\n",
       "      <th>alignment_algorithm</th>\n",
       "      <th>variation_calling_algorithm</th>\n",
       "      <th>other_analysis_algorithm</th>\n",
       "      <th>seq_coverage</th>\n",
       "      <th>raw_data_repository</th>\n",
       "      <th>raw_data_accession</th>\n",
       "      <th>initial_data_release_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60228</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A12B-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WXS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-MC3 https://gdc.cancer.gov/about-data/pub...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GDC</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60228</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A12B-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WXS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-MC3 https://gdc.cancer.gov/about-data/pub...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GDC</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60228</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A12B-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WXS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-MC3 https://gdc.cancer.gov/about-data/pub...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GDC</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60252</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A19H-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A19H-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG Consensus SNV-MNV caller</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FI30202:FI30201</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60252</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A19H-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A19H-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG Consensus SNV-MNV caller</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FI30202:FI30201</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60252</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A19H-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A19H-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG Consensus SNV-MNV caller</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FI30202:FI30201</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_mutation_id icgc_donor_id project_code icgc_specimen_id icgc_sample_id  \\\n",
       "4          MU23549        DO5200      BRCA-US          SP10944        SA60228   \n",
       "5          MU23549        DO5200      BRCA-US          SP10944        SA60228   \n",
       "6          MU23549        DO5200      BRCA-US          SP10944        SA60228   \n",
       "7          MU23549        DO5200      BRCA-US          SP10944        SA60252   \n",
       "8          MU23549        DO5200      BRCA-US          SP10944        SA60252   \n",
       "9          MU23549        DO5200      BRCA-US          SP10944        SA60252   \n",
       "\n",
       "  matched_icgc_sample_id           submitted_sample_id  \\\n",
       "4                SA60324  TCGA-BH-A18U-01A-21D-A12B-09   \n",
       "5                SA60324  TCGA-BH-A18U-01A-21D-A12B-09   \n",
       "6                SA60324  TCGA-BH-A18U-01A-21D-A12B-09   \n",
       "7                SA60348  TCGA-BH-A18U-01A-21D-A19H-09   \n",
       "8                SA60348  TCGA-BH-A18U-01A-21D-A19H-09   \n",
       "9                SA60348  TCGA-BH-A18U-01A-21D-A19H-09   \n",
       "\n",
       "    submitted_matched_sample_id chromosome  chromosome_start  ...  \\\n",
       "4  TCGA-BH-A18U-11A-23D-A12B-09          7          70886058  ...   \n",
       "5  TCGA-BH-A18U-11A-23D-A12B-09          7          70886058  ...   \n",
       "6  TCGA-BH-A18U-11A-23D-A12B-09          7          70886058  ...   \n",
       "7  TCGA-BH-A18U-11A-23D-A19H-09          7          70886058  ...   \n",
       "8  TCGA-BH-A18U-11A-23D-A19H-09          7          70886058  ...   \n",
       "9  TCGA-BH-A18U-11A-23D-A19H-09          7          70886058  ...   \n",
       "\n",
       "   experimental_protocol  sequencing_strategy base_calling_algorithm  \\\n",
       "4                    NaN                  WXS                    NaN   \n",
       "5                    NaN                  WXS                    NaN   \n",
       "6                    NaN                  WXS                    NaN   \n",
       "7                    NaN                  WGS                    NaN   \n",
       "8                    NaN                  WGS                    NaN   \n",
       "9                    NaN                  WGS                    NaN   \n",
       "\n",
       "  alignment_algorithm                        variation_calling_algorithm  \\\n",
       "4                 NaN  TCGA-MC3 https://gdc.cancer.gov/about-data/pub...   \n",
       "5                 NaN  TCGA-MC3 https://gdc.cancer.gov/about-data/pub...   \n",
       "6                 NaN  TCGA-MC3 https://gdc.cancer.gov/about-data/pub...   \n",
       "7                 NaN                     PCAWG Consensus SNV-MNV caller   \n",
       "8                 NaN                     PCAWG Consensus SNV-MNV caller   \n",
       "9                 NaN                     PCAWG Consensus SNV-MNV caller   \n",
       "\n",
       "  other_analysis_algorithm seq_coverage  raw_data_repository  \\\n",
       "4                      NaN          NaN                  GDC   \n",
       "5                      NaN          NaN                  GDC   \n",
       "6                      NaN          NaN                  GDC   \n",
       "7                      NaN          NaN                  NaN   \n",
       "8                      NaN          NaN                  NaN   \n",
       "9                      NaN          NaN                  NaN   \n",
       "\n",
       "             raw_data_accession  initial_data_release_date  \n",
       "4  TCGA-BH-A18U-01A-21D-A12B-09                        NaN  \n",
       "5  TCGA-BH-A18U-01A-21D-A12B-09                        NaN  \n",
       "6  TCGA-BH-A18U-01A-21D-A12B-09                        NaN  \n",
       "7               FI30202:FI30201                        NaN  \n",
       "8               FI30202:FI30201                        NaN  \n",
       "9               FI30202:FI30201                        NaN  \n",
       "\n",
       "[6 rows x 42 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Mutation ID MU23549\n",
    "df[df['icgc_mutation_id'] == 'MU23549']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For the files I looked at (which was only a few so far) there are 42 features."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Some EDA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['icgc_mutation_id', 'icgc_donor_id', 'project_code', 'icgc_specimen_id',\n",
       "       'icgc_sample_id', 'matched_icgc_sample_id', 'submitted_sample_id',\n",
       "       'submitted_matched_sample_id', 'chromosome', 'chromosome_start',\n",
       "       'chromosome_end', 'chromosome_strand', 'assembly_version',\n",
       "       'mutation_type', 'reference_genome_allele', 'mutated_from_allele',\n",
       "       'mutated_to_allele', 'quality_score', 'probability', 'total_read_count',\n",
       "       'mutant_allele_read_count', 'verification_status',\n",
       "       'verification_platform', 'biological_validation_status',\n",
       "       'biological_validation_platform', 'consequence_type', 'aa_mutation',\n",
       "       'cds_mutation', 'gene_affected', 'transcript_affected',\n",
       "       'gene_build_version', 'platform', 'experimental_protocol',\n",
       "       'sequencing_strategy', 'base_calling_algorithm', 'alignment_algorithm',\n",
       "       'variation_calling_algorithm', 'other_analysis_algorithm',\n",
       "       'seq_coverage', 'raw_data_repository', 'raw_data_accession',\n",
       "       'initial_data_release_date'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Choices of features\n",
    "- Chromosome start and end may not be helpful though there could exist a relationship between which chromosome and the position of mutation. \n",
    "- Do not know what CDS mutation or AA mutation is. Many NaNs here.\n",
    "- Do not know how the gene_effected feature is encoded.\n",
    "- Project_code can be the target output.\n",
    "- Possibly remove start and end features with length of mutation. If a deletion then we can represent with negative number."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>reference_genome_allele</th>\n",
       "      <th>mutated_from_allele</th>\n",
       "      <th>mutated_to_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2192</th>\n",
       "      <td>3</td>\n",
       "      <td>48665399</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2193</th>\n",
       "      <td>3</td>\n",
       "      <td>48665399</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2194</th>\n",
       "      <td>3</td>\n",
       "      <td>48665399</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2195</th>\n",
       "      <td>3</td>\n",
       "      <td>48665399</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2196</th>\n",
       "      <td>3</td>\n",
       "      <td>48665399</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2197 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chromosome  chromosome_start reference_genome_allele mutated_from_allele  \\\n",
       "0             5          58021869                       C                   C   \n",
       "1             5          58021869                       C                   C   \n",
       "2             5          58021869                       C                   C   \n",
       "3             5          58021869                       C                   C   \n",
       "4             7          70886058                       G                   G   \n",
       "...         ...               ...                     ...                 ...   \n",
       "2192          3          48665399                       C                   C   \n",
       "2193          3          48665399                       C                   C   \n",
       "2194          3          48665399                       C                   C   \n",
       "2195          3          48665399                       C                   C   \n",
       "2196          3          48665399                       C                   C   \n",
       "\n",
       "     mutated_to_allele  \n",
       "0                    A  \n",
       "1                    A  \n",
       "2                    A  \n",
       "3                    A  \n",
       "4                    A  \n",
       "...                ...  \n",
       "2192                 T  \n",
       "2193                 T  \n",
       "2194                 T  \n",
       "2195                 T  \n",
       "2196                 T  \n",
       "\n",
       "[2197 rows x 5 columns]"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Choice of features here is due to quick look at the columns. \n",
    "# Change as you see fit.\n",
    "\n",
    "features = ['icgc_mutation_id', \n",
    "            'icgc_donor_id',\n",
    "            'icgc_sample_id', \n",
    "            'matched_icgc_sample_id', \n",
    "            'submitted_sample_id',\n",
    "            'submitted_matched_sample_id',\n",
    "            'chromosome', \n",
    "            'chromosome_start',\n",
    "            'chromosome_end',\n",
    "            'chromosome_strand',\n",
    "            'reference_genome_allele',\n",
    "            'mutated_from_allele',\n",
    "            'mutated_to_allele',\n",
    "            'consequence_type',\n",
    "            'gene_affected',\n",
    "            'total_read_count',\n",
    "            'project_code']\n",
    "\n",
    "np.unique(df[features].chromosome_strand)\n",
    "df[features][['chromosome', \n",
    "              'chromosome_start', \n",
    "              'reference_genome_allele', \n",
    "              'mutated_from_allele', \n",
    "              'mutated_to_allele']]\n",
    "\n",
    "# why would donor have two submitted samples or more?\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mutation IDs\n",
    "- There are many duplicates. There are only 213 unique mutation IDs for this donor."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 213 unique mutation IDs\n"
     ]
    }
   ],
   "source": [
    "print('There are {} unique mutation IDs'.format(len(set(df[features]['icgc_mutation_id']))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There is only 1 donor in this file\n"
     ]
    }
   ],
   "source": [
    "print('There is only {} donor in this file'.format(len(set(df[features]['icgc_donor_id']))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = df[features]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_mutation_id</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_specimen_id</th>\n",
       "      <th>matched_icgc_sample_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>chromosome_end</th>\n",
       "      <th>chromosome_strand</th>\n",
       "      <th>assembly_version</th>\n",
       "      <th>...</th>\n",
       "      <th>probability</th>\n",
       "      <th>total_read_count</th>\n",
       "      <th>mutant_allele_read_count</th>\n",
       "      <th>verification_status</th>\n",
       "      <th>consequence_type</th>\n",
       "      <th>aa_mutation</th>\n",
       "      <th>cds_mutation</th>\n",
       "      <th>gene_affected</th>\n",
       "      <th>transcript_affected</th>\n",
       "      <th>gene_build_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>58021869</td>\n",
       "      <td>1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68</td>\n",
       "      <td>12</td>\n",
       "      <td>not tested</td>\n",
       "      <td>exon_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>ENST00000507977</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>58021869</td>\n",
       "      <td>1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68</td>\n",
       "      <td>12</td>\n",
       "      <td>not tested</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A98D</td>\n",
       "      <td>293C&gt;A</td>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>ENST00000282878</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>58021869</td>\n",
       "      <td>1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>349</td>\n",
       "      <td>49</td>\n",
       "      <td>not tested</td>\n",
       "      <td>exon_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>ENST00000507977</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>58021869</td>\n",
       "      <td>1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>349</td>\n",
       "      <td>49</td>\n",
       "      <td>not tested</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A98D</td>\n",
       "      <td>293C&gt;A</td>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>ENST00000282878</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>70886058</td>\n",
       "      <td>1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>53</td>\n",
       "      <td>37</td>\n",
       "      <td>not tested</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>W310*</td>\n",
       "      <td>929G&gt;A</td>\n",
       "      <td>ENSG00000185274</td>\n",
       "      <td>ENST00000333538</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_mutation_id icgc_donor_id project_code icgc_specimen_id  \\\n",
       "0          MU22195        DO5200      BRCA-US          SP10944   \n",
       "1          MU22195        DO5200      BRCA-US          SP10944   \n",
       "2          MU22195        DO5200      BRCA-US          SP10944   \n",
       "3          MU22195        DO5200      BRCA-US          SP10944   \n",
       "4          MU23549        DO5200      BRCA-US          SP10944   \n",
       "\n",
       "  matched_icgc_sample_id chromosome  chromosome_start  chromosome_end  \\\n",
       "0                SA60348          5          58021869        58021869   \n",
       "1                SA60348          5          58021869        58021869   \n",
       "2                SA60324          5          58021869        58021869   \n",
       "3                SA60324          5          58021869        58021869   \n",
       "4                SA60324          7          70886058        70886058   \n",
       "\n",
       "   chromosome_strand assembly_version  ... probability total_read_count  \\\n",
       "0                  1           GRCh37  ...         NaN               68   \n",
       "1                  1           GRCh37  ...         NaN               68   \n",
       "2                  1           GRCh37  ...         NaN              349   \n",
       "3                  1           GRCh37  ...         NaN              349   \n",
       "4                  1           GRCh37  ...         NaN               53   \n",
       "\n",
       "  mutant_allele_read_count verification_status  consequence_type  aa_mutation  \\\n",
       "0                       12          not tested      exon_variant          NaN   \n",
       "1                       12          not tested  missense_variant         A98D   \n",
       "2                       49          not tested      exon_variant          NaN   \n",
       "3                       49          not tested  missense_variant         A98D   \n",
       "4                       37          not tested       stop_gained        W310*   \n",
       "\n",
       "   cds_mutation    gene_affected transcript_affected gene_build_version  \n",
       "0           NaN  ENSG00000152932     ENST00000507977                 75  \n",
       "1        293C>A  ENSG00000152932     ENST00000282878                 75  \n",
       "2           NaN  ENSG00000152932     ENST00000507977                 75  \n",
       "3        293C>A  ENSG00000152932     ENST00000282878                 75  \n",
       "4        929G>A  ENSG00000185274     ENST00000333538                 75  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Not all mutations are 1bp\n",
    "Some mutations involve longer sequences!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mutation_type</th>\n",
       "      <th>reference_genome_allele</th>\n",
       "      <th>mutated_from_allele</th>\n",
       "      <th>mutated_to_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>deletion of &lt;=200bp</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>deletion of &lt;=200bp</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>deletion of &lt;=200bp</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>deletion of &lt;=200bp</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>deletion of &lt;=200bp</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           mutation_type reference_genome_allele  mutated_from_allele  \\\n",
       "159  deletion of <=200bp     TGCGGAGATTCTCTTCCTC  TGCGGAGATTCTCTTCCTC   \n",
       "160  deletion of <=200bp     TGCGGAGATTCTCTTCCTC  TGCGGAGATTCTCTTCCTC   \n",
       "161  deletion of <=200bp     TGCGGAGATTCTCTTCCTC  TGCGGAGATTCTCTTCCTC   \n",
       "162  deletion of <=200bp     TGCGGAGATTCTCTTCCTC  TGCGGAGATTCTCTTCCTC   \n",
       "163  deletion of <=200bp     TGCGGAGATTCTCTTCCTC  TGCGGAGATTCTCTTCCTC   \n",
       "\n",
       "    mutated_to_allele  \n",
       "159                 -  \n",
       "160                 -  \n",
       "161                 -  \n",
       "162                 -  \n",
       "163                 -  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['mutation_type'] == 'deletion of <=200bp'][['mutation_type', 'reference_genome_allele', 'mutated_from_allele', 'mutated_to_allele']].head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'3_prime_UTR_variant',\n",
       " '5_prime_UTR_premature_start_codon_gain_variant',\n",
       " '5_prime_UTR_variant',\n",
       " 'disruptive_inframe_deletion',\n",
       " 'downstream_gene_variant',\n",
       " 'exon_variant',\n",
       " 'frameshift_variant',\n",
       " 'inframe_deletion',\n",
       " 'intron_variant',\n",
       " 'missense_variant',\n",
       " 'splice_acceptor_variant',\n",
       " 'splice_donor_variant',\n",
       " 'splice_region_variant',\n",
       " 'stop_gained',\n",
       " 'synonymous_variant',\n",
       " 'upstream_gene_variant'}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(df['consequence_type'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
