{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from os import listdir\n",
    "from statistics import mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['OV-US',\n",
       " 'organize_data.py',\n",
       " 'COAD-US',\n",
       " '.DS_Store',\n",
       " 'list_data_dirs.py',\n",
       " 'UCEC-US',\n",
       " 'PRAD-US',\n",
       " 'KIRC-US',\n",
       " 'SKCM-US',\n",
       " 'THCA-US',\n",
       " 'LGG-US',\n",
       " 'GBM-US',\n",
       " 'LUSC-US',\n",
       " 'BLCA-US',\n",
       " 'BRCA-US',\n",
       " '.remove_duplicates.py.swo']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "listdir('data')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "def look(cancer_code):\n",
    "    DATA_PATH = 'data/'\n",
    "    PATH = DATA_PATH + cancer_code + '/'\n",
    "    \n",
    "    feat = []\n",
    "    ids = []\n",
    "    donors = []\n",
    "    \n",
    "    all_files = [file for file in listdir(PATH)]\n",
    "    \n",
    "    for file in all_files:\n",
    "        df = pd.read_csv(PATH + file, sep='\\t')\n",
    "        feat.append(len(df.columns))\n",
    "        ids.append(len(set(df['icgc_mutation_id'])))\n",
    "        donors.append(len(set(df['icgc_donor_id'])))\n",
    "    \n",
    "    print('*' * 40)\n",
    "    print('Project code: ', cancer_code)\n",
    "    \n",
    "    try:\n",
    "        print('Average number of features: {}'.format(mean(feat)))\n",
    "        print('Average number of mutations: {}'.format(mean(ids)))\n",
    "        print('Number of donors: {}'.format(sum(donors)))\n",
    "    except:\n",
    "        print(NULL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "****************************************\n",
      "Project code:  OV-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 105.53217821782178\n",
      "Number of donors: 404\n",
      "****************************************\n",
      "Project code:  COAD-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 1949.2\n",
      "Number of donors: 280\n",
      "****************************************\n",
      "Project code:  UCEC-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 3093.343108504399\n",
      "Number of donors: 341\n",
      "****************************************\n",
      "Project code:  PRAD-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 25.90277777777778\n",
      "Number of donors: 144\n",
      "****************************************\n",
      "Project code:  KIRC-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 106.89270386266094\n",
      "Number of donors: 233\n",
      "****************************************\n",
      "Project code:  SKCM-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 1890.2366666666667\n",
      "Number of donors: 300\n",
      "****************************************\n",
      "Project code:  THCA-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 41.88135593220339\n",
      "Number of donors: 295\n",
      "****************************************\n",
      "Project code:  LGG-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 175.17207792207793\n",
      "Number of donors: 308\n",
      "****************************************\n",
      "Project code:  GBM-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 188.38186813186815\n",
      "Number of donors: 364\n",
      "****************************************\n",
      "Project code:  LUSC-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 9.411157024793388\n",
      "Number of donors: 484\n",
      "****************************************\n",
      "Project code:  BLCA-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 55.48048780487805\n",
      "Number of donors: 410\n",
      "****************************************\n",
      "Project code:  BRCA-US\n",
      "Average number of features: 42\n",
      "Average number of mutations: 130.88801571709234\n",
      "Number of donors: 1018\n"
     ]
    }
   ],
   "source": [
    "folders = [file for file in listdir('data') if 'US' in file]\n",
    "for cancer in folders:\n",
    "    look(cancer)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reading in each file\n",
    "Each file represents one individual. We could probably use the project_code feature as our target/output."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/BRCA-US/simple_somatic_mutation.open-2020-03-02T154602.752.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of features: 42\n",
      "Index(['icgc_mutation_id', 'icgc_donor_id', 'project_code', 'icgc_specimen_id',\n",
      "       'icgc_sample_id', 'matched_icgc_sample_id', 'submitted_sample_id',\n",
      "       'submitted_matched_sample_id', 'chromosome', 'chromosome_start',\n",
      "       'chromosome_end', 'chromosome_strand', 'assembly_version',\n",
      "       'mutation_type', 'reference_genome_allele', 'mutated_from_allele',\n",
      "       'mutated_to_allele', 'quality_score', 'probability', 'total_read_count',\n",
      "       'mutant_allele_read_count', 'verification_status',\n",
      "       'verification_platform', 'biological_validation_status',\n",
      "       'biological_validation_platform', 'consequence_type', 'aa_mutation',\n",
      "       'cds_mutation', 'gene_affected', 'transcript_affected',\n",
      "       'gene_build_version', 'platform', 'experimental_protocol',\n",
      "       'sequencing_strategy', 'base_calling_algorithm', 'alignment_algorithm',\n",
      "       'variation_calling_algorithm', 'other_analysis_algorithm',\n",
      "       'seq_coverage', 'raw_data_repository', 'raw_data_accession',\n",
      "       'initial_data_release_date'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "print('Number of features: {}'.format(len(df.columns)))\n",
    "print(df.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_mutation_id</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_specimen_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>matched_icgc_sample_id</th>\n",
       "      <th>submitted_sample_id</th>\n",
       "      <th>submitted_matched_sample_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>...</th>\n",
       "      <th>experimental_protocol</th>\n",
       "      <th>sequencing_strategy</th>\n",
       "      <th>base_calling_algorithm</th>\n",
       "      <th>alignment_algorithm</th>\n",
       "      <th>variation_calling_algorithm</th>\n",
       "      <th>other_analysis_algorithm</th>\n",
       "      <th>seq_coverage</th>\n",
       "      <th>raw_data_repository</th>\n",
       "      <th>raw_data_accession</th>\n",
       "      <th>initial_data_release_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60252</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A19H-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A19H-09</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG Consensus SNV-MNV caller</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FI30202:FI30201</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60252</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A19H-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A19H-09</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG Consensus SNV-MNV caller</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FI30202:FI30201</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60228</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A12B-09</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WXS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-MC3 https://gdc.cancer.gov/about-data/pub...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GDC</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60228</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A12B-09</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WXS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-MC3 https://gdc.cancer.gov/about-data/pub...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GDC</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60228</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A12B-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WXS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-MC3 https://gdc.cancer.gov/about-data/pub...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GDC</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_mutation_id icgc_donor_id project_code icgc_specimen_id icgc_sample_id  \\\n",
       "0          MU22195        DO5200      BRCA-US          SP10944        SA60252   \n",
       "1          MU22195        DO5200      BRCA-US          SP10944        SA60252   \n",
       "2          MU22195        DO5200      BRCA-US          SP10944        SA60228   \n",
       "3          MU22195        DO5200      BRCA-US          SP10944        SA60228   \n",
       "4          MU23549        DO5200      BRCA-US          SP10944        SA60228   \n",
       "\n",
       "  matched_icgc_sample_id           submitted_sample_id  \\\n",
       "0                SA60348  TCGA-BH-A18U-01A-21D-A19H-09   \n",
       "1                SA60348  TCGA-BH-A18U-01A-21D-A19H-09   \n",
       "2                SA60324  TCGA-BH-A18U-01A-21D-A12B-09   \n",
       "3                SA60324  TCGA-BH-A18U-01A-21D-A12B-09   \n",
       "4                SA60324  TCGA-BH-A18U-01A-21D-A12B-09   \n",
       "\n",
       "    submitted_matched_sample_id chromosome  chromosome_start  ...  \\\n",
       "0  TCGA-BH-A18U-11A-23D-A19H-09          5          58021869  ...   \n",
       "1  TCGA-BH-A18U-11A-23D-A19H-09          5          58021869  ...   \n",
       "2  TCGA-BH-A18U-11A-23D-A12B-09          5          58021869  ...   \n",
       "3  TCGA-BH-A18U-11A-23D-A12B-09          5          58021869  ...   \n",
       "4  TCGA-BH-A18U-11A-23D-A12B-09          7          70886058  ...   \n",
       "\n",
       "   experimental_protocol  sequencing_strategy base_calling_algorithm  \\\n",
       "0                    NaN                  WGS                    NaN   \n",
       "1                    NaN                  WGS                    NaN   \n",
       "2                    NaN                  WXS                    NaN   \n",
       "3                    NaN                  WXS                    NaN   \n",
       "4                    NaN                  WXS                    NaN   \n",
       "\n",
       "  alignment_algorithm                        variation_calling_algorithm  \\\n",
       "0                 NaN                     PCAWG Consensus SNV-MNV caller   \n",
       "1                 NaN                     PCAWG Consensus SNV-MNV caller   \n",
       "2                 NaN  TCGA-MC3 https://gdc.cancer.gov/about-data/pub...   \n",
       "3                 NaN  TCGA-MC3 https://gdc.cancer.gov/about-data/pub...   \n",
       "4                 NaN  TCGA-MC3 https://gdc.cancer.gov/about-data/pub...   \n",
       "\n",
       "  other_analysis_algorithm seq_coverage  raw_data_repository  \\\n",
       "0                      NaN          NaN                  NaN   \n",
       "1                      NaN          NaN                  NaN   \n",
       "2                      NaN          NaN                  GDC   \n",
       "3                      NaN          NaN                  GDC   \n",
       "4                      NaN          NaN                  GDC   \n",
       "\n",
       "             raw_data_accession  initial_data_release_date  \n",
       "0               FI30202:FI30201                        NaN  \n",
       "1               FI30202:FI30201                        NaN  \n",
       "2  TCGA-BH-A18U-01A-21D-A12B-09                        NaN  \n",
       "3  TCGA-BH-A18U-01A-21D-A12B-09                        NaN  \n",
       "4  TCGA-BH-A18U-01A-21D-A12B-09                        NaN  \n",
       "\n",
       "[5 rows x 42 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notice how for one donor there are duplicate icgc_mutation_id values. Not sure what this means."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_mutation_id</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_specimen_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>matched_icgc_sample_id</th>\n",
       "      <th>submitted_sample_id</th>\n",
       "      <th>submitted_matched_sample_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>...</th>\n",
       "      <th>experimental_protocol</th>\n",
       "      <th>sequencing_strategy</th>\n",
       "      <th>base_calling_algorithm</th>\n",
       "      <th>alignment_algorithm</th>\n",
       "      <th>variation_calling_algorithm</th>\n",
       "      <th>other_analysis_algorithm</th>\n",
       "      <th>seq_coverage</th>\n",
       "      <th>raw_data_repository</th>\n",
       "      <th>raw_data_accession</th>\n",
       "      <th>initial_data_release_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60252</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A19H-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A19H-09</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG Consensus SNV-MNV caller</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FI30202:FI30201</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60252</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A19H-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A19H-09</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG Consensus SNV-MNV caller</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FI30202:FI30201</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60228</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A12B-09</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WXS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-MC3 https://gdc.cancer.gov/about-data/pub...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GDC</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60228</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A12B-09</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WXS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-MC3 https://gdc.cancer.gov/about-data/pub...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GDC</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_mutation_id icgc_donor_id project_code icgc_specimen_id icgc_sample_id  \\\n",
       "0          MU22195        DO5200      BRCA-US          SP10944        SA60252   \n",
       "1          MU22195        DO5200      BRCA-US          SP10944        SA60252   \n",
       "2          MU22195        DO5200      BRCA-US          SP10944        SA60228   \n",
       "3          MU22195        DO5200      BRCA-US          SP10944        SA60228   \n",
       "\n",
       "  matched_icgc_sample_id           submitted_sample_id  \\\n",
       "0                SA60348  TCGA-BH-A18U-01A-21D-A19H-09   \n",
       "1                SA60348  TCGA-BH-A18U-01A-21D-A19H-09   \n",
       "2                SA60324  TCGA-BH-A18U-01A-21D-A12B-09   \n",
       "3                SA60324  TCGA-BH-A18U-01A-21D-A12B-09   \n",
       "\n",
       "    submitted_matched_sample_id chromosome  chromosome_start  ...  \\\n",
       "0  TCGA-BH-A18U-11A-23D-A19H-09          5          58021869  ...   \n",
       "1  TCGA-BH-A18U-11A-23D-A19H-09          5          58021869  ...   \n",
       "2  TCGA-BH-A18U-11A-23D-A12B-09          5          58021869  ...   \n",
       "3  TCGA-BH-A18U-11A-23D-A12B-09          5          58021869  ...   \n",
       "\n",
       "   experimental_protocol  sequencing_strategy base_calling_algorithm  \\\n",
       "0                    NaN                  WGS                    NaN   \n",
       "1                    NaN                  WGS                    NaN   \n",
       "2                    NaN                  WXS                    NaN   \n",
       "3                    NaN                  WXS                    NaN   \n",
       "\n",
       "  alignment_algorithm                        variation_calling_algorithm  \\\n",
       "0                 NaN                     PCAWG Consensus SNV-MNV caller   \n",
       "1                 NaN                     PCAWG Consensus SNV-MNV caller   \n",
       "2                 NaN  TCGA-MC3 https://gdc.cancer.gov/about-data/pub...   \n",
       "3                 NaN  TCGA-MC3 https://gdc.cancer.gov/about-data/pub...   \n",
       "\n",
       "  other_analysis_algorithm seq_coverage  raw_data_repository  \\\n",
       "0                      NaN          NaN                  NaN   \n",
       "1                      NaN          NaN                  NaN   \n",
       "2                      NaN          NaN                  GDC   \n",
       "3                      NaN          NaN                  GDC   \n",
       "\n",
       "             raw_data_accession  initial_data_release_date  \n",
       "0               FI30202:FI30201                        NaN  \n",
       "1               FI30202:FI30201                        NaN  \n",
       "2  TCGA-BH-A18U-01A-21D-A12B-09                        NaN  \n",
       "3  TCGA-BH-A18U-01A-21D-A12B-09                        NaN  \n",
       "\n",
       "[4 rows x 42 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Mutation ID MU22195\n",
    "df[df['icgc_mutation_id'] == 'MU22195']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_mutation_id</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_specimen_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>matched_icgc_sample_id</th>\n",
       "      <th>submitted_sample_id</th>\n",
       "      <th>submitted_matched_sample_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>...</th>\n",
       "      <th>experimental_protocol</th>\n",
       "      <th>sequencing_strategy</th>\n",
       "      <th>base_calling_algorithm</th>\n",
       "      <th>alignment_algorithm</th>\n",
       "      <th>variation_calling_algorithm</th>\n",
       "      <th>other_analysis_algorithm</th>\n",
       "      <th>seq_coverage</th>\n",
       "      <th>raw_data_repository</th>\n",
       "      <th>raw_data_accession</th>\n",
       "      <th>initial_data_release_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60228</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A12B-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WXS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-MC3 https://gdc.cancer.gov/about-data/pub...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GDC</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60228</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A12B-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WXS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-MC3 https://gdc.cancer.gov/about-data/pub...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GDC</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60228</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A12B-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WXS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-MC3 https://gdc.cancer.gov/about-data/pub...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GDC</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A12B-09</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60252</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A19H-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A19H-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG Consensus SNV-MNV caller</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FI30202:FI30201</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60252</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A19H-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A19H-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG Consensus SNV-MNV caller</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FI30202:FI30201</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60252</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>TCGA-BH-A18U-01A-21D-A19H-09</td>\n",
       "      <td>TCGA-BH-A18U-11A-23D-A19H-09</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WGS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PCAWG Consensus SNV-MNV caller</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FI30202:FI30201</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_mutation_id icgc_donor_id project_code icgc_specimen_id icgc_sample_id  \\\n",
       "4          MU23549        DO5200      BRCA-US          SP10944        SA60228   \n",
       "5          MU23549        DO5200      BRCA-US          SP10944        SA60228   \n",
       "6          MU23549        DO5200      BRCA-US          SP10944        SA60228   \n",
       "7          MU23549        DO5200      BRCA-US          SP10944        SA60252   \n",
       "8          MU23549        DO5200      BRCA-US          SP10944        SA60252   \n",
       "9          MU23549        DO5200      BRCA-US          SP10944        SA60252   \n",
       "\n",
       "  matched_icgc_sample_id           submitted_sample_id  \\\n",
       "4                SA60324  TCGA-BH-A18U-01A-21D-A12B-09   \n",
       "5                SA60324  TCGA-BH-A18U-01A-21D-A12B-09   \n",
       "6                SA60324  TCGA-BH-A18U-01A-21D-A12B-09   \n",
       "7                SA60348  TCGA-BH-A18U-01A-21D-A19H-09   \n",
       "8                SA60348  TCGA-BH-A18U-01A-21D-A19H-09   \n",
       "9                SA60348  TCGA-BH-A18U-01A-21D-A19H-09   \n",
       "\n",
       "    submitted_matched_sample_id chromosome  chromosome_start  ...  \\\n",
       "4  TCGA-BH-A18U-11A-23D-A12B-09          7          70886058  ...   \n",
       "5  TCGA-BH-A18U-11A-23D-A12B-09          7          70886058  ...   \n",
       "6  TCGA-BH-A18U-11A-23D-A12B-09          7          70886058  ...   \n",
       "7  TCGA-BH-A18U-11A-23D-A19H-09          7          70886058  ...   \n",
       "8  TCGA-BH-A18U-11A-23D-A19H-09          7          70886058  ...   \n",
       "9  TCGA-BH-A18U-11A-23D-A19H-09          7          70886058  ...   \n",
       "\n",
       "   experimental_protocol  sequencing_strategy base_calling_algorithm  \\\n",
       "4                    NaN                  WXS                    NaN   \n",
       "5                    NaN                  WXS                    NaN   \n",
       "6                    NaN                  WXS                    NaN   \n",
       "7                    NaN                  WGS                    NaN   \n",
       "8                    NaN                  WGS                    NaN   \n",
       "9                    NaN                  WGS                    NaN   \n",
       "\n",
       "  alignment_algorithm                        variation_calling_algorithm  \\\n",
       "4                 NaN  TCGA-MC3 https://gdc.cancer.gov/about-data/pub...   \n",
       "5                 NaN  TCGA-MC3 https://gdc.cancer.gov/about-data/pub...   \n",
       "6                 NaN  TCGA-MC3 https://gdc.cancer.gov/about-data/pub...   \n",
       "7                 NaN                     PCAWG Consensus SNV-MNV caller   \n",
       "8                 NaN                     PCAWG Consensus SNV-MNV caller   \n",
       "9                 NaN                     PCAWG Consensus SNV-MNV caller   \n",
       "\n",
       "  other_analysis_algorithm seq_coverage  raw_data_repository  \\\n",
       "4                      NaN          NaN                  GDC   \n",
       "5                      NaN          NaN                  GDC   \n",
       "6                      NaN          NaN                  GDC   \n",
       "7                      NaN          NaN                  NaN   \n",
       "8                      NaN          NaN                  NaN   \n",
       "9                      NaN          NaN                  NaN   \n",
       "\n",
       "             raw_data_accession  initial_data_release_date  \n",
       "4  TCGA-BH-A18U-01A-21D-A12B-09                        NaN  \n",
       "5  TCGA-BH-A18U-01A-21D-A12B-09                        NaN  \n",
       "6  TCGA-BH-A18U-01A-21D-A12B-09                        NaN  \n",
       "7               FI30202:FI30201                        NaN  \n",
       "8               FI30202:FI30201                        NaN  \n",
       "9               FI30202:FI30201                        NaN  \n",
       "\n",
       "[6 rows x 42 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Mutation ID MU23549\n",
    "df[df['icgc_mutation_id'] == 'MU23549']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For the files I looked at (which was only a few so far) there are 42 features."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Some EDA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['icgc_mutation_id', 'icgc_donor_id', 'project_code', 'icgc_specimen_id',\n",
       "       'icgc_sample_id', 'matched_icgc_sample_id', 'submitted_sample_id',\n",
       "       'submitted_matched_sample_id', 'chromosome', 'chromosome_start',\n",
       "       'chromosome_end', 'chromosome_strand', 'assembly_version',\n",
       "       'mutation_type', 'reference_genome_allele', 'mutated_from_allele',\n",
       "       'mutated_to_allele', 'quality_score', 'probability', 'total_read_count',\n",
       "       'mutant_allele_read_count', 'verification_status',\n",
       "       'verification_platform', 'biological_validation_status',\n",
       "       'biological_validation_platform', 'consequence_type', 'aa_mutation',\n",
       "       'cds_mutation', 'gene_affected', 'transcript_affected',\n",
       "       'gene_build_version', 'platform', 'experimental_protocol',\n",
       "       'sequencing_strategy', 'base_calling_algorithm', 'alignment_algorithm',\n",
       "       'variation_calling_algorithm', 'other_analysis_algorithm',\n",
       "       'seq_coverage', 'raw_data_repository', 'raw_data_accession',\n",
       "       'initial_data_release_date'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Choices of features\n",
    "- Chromosome start and end may not be helpful though there could exist a relationship between which chromosome and the position of mutation. \n",
    "- Do not know what CDS mutation or AA mutation is. Many NaNs here.\n",
    "- Do not know how the gene_effected feature is encoded.\n",
    "- Project_code can be the target output.\n",
    "- Possibly remove start and end features with length of mutation. If a deletion then we can represent with negative number."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_mutation_id</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>chromosome_end</th>\n",
       "      <th>chromosome_strand</th>\n",
       "      <th>reference_genome_allele</th>\n",
       "      <th>mutated_from_allele</th>\n",
       "      <th>mutated_to_allele</th>\n",
       "      <th>consequence_type</th>\n",
       "      <th>gene_affected</th>\n",
       "      <th>total_read_count</th>\n",
       "      <th>project_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>58021869</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exon_variant</td>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>68</td>\n",
       "      <td>BRCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>58021869</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>68</td>\n",
       "      <td>BRCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>58021869</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exon_variant</td>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>349</td>\n",
       "      <td>BRCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>58021869</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>349</td>\n",
       "      <td>BRCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>70886058</td>\n",
       "      <td>1</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>ENSG00000185274</td>\n",
       "      <td>53</td>\n",
       "      <td>BRCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2192</th>\n",
       "      <td>MU79396</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>3</td>\n",
       "      <td>48665399</td>\n",
       "      <td>48665399</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>ENSG00000225697</td>\n",
       "      <td>292</td>\n",
       "      <td>BRCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2193</th>\n",
       "      <td>MU79396</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>3</td>\n",
       "      <td>48665399</td>\n",
       "      <td>48665399</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>ENSG00000225697</td>\n",
       "      <td>292</td>\n",
       "      <td>BRCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2194</th>\n",
       "      <td>MU79396</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>3</td>\n",
       "      <td>48665399</td>\n",
       "      <td>48665399</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>ENSG00000225697</td>\n",
       "      <td>292</td>\n",
       "      <td>BRCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2195</th>\n",
       "      <td>MU79396</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>3</td>\n",
       "      <td>48665399</td>\n",
       "      <td>48665399</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>ENSG00000225697</td>\n",
       "      <td>292</td>\n",
       "      <td>BRCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2196</th>\n",
       "      <td>MU79396</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>3</td>\n",
       "      <td>48665399</td>\n",
       "      <td>48665399</td>\n",
       "      <td>1</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>ENSG00000225697</td>\n",
       "      <td>292</td>\n",
       "      <td>BRCA-US</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2197 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     icgc_mutation_id icgc_donor_id chromosome  chromosome_start  \\\n",
       "0             MU22195        DO5200          5          58021869   \n",
       "1             MU22195        DO5200          5          58021869   \n",
       "2             MU22195        DO5200          5          58021869   \n",
       "3             MU22195        DO5200          5          58021869   \n",
       "4             MU23549        DO5200          7          70886058   \n",
       "...               ...           ...        ...               ...   \n",
       "2192          MU79396        DO5200          3          48665399   \n",
       "2193          MU79396        DO5200          3          48665399   \n",
       "2194          MU79396        DO5200          3          48665399   \n",
       "2195          MU79396        DO5200          3          48665399   \n",
       "2196          MU79396        DO5200          3          48665399   \n",
       "\n",
       "      chromosome_end  chromosome_strand reference_genome_allele  \\\n",
       "0           58021869                  1                       C   \n",
       "1           58021869                  1                       C   \n",
       "2           58021869                  1                       C   \n",
       "3           58021869                  1                       C   \n",
       "4           70886058                  1                       G   \n",
       "...              ...                ...                     ...   \n",
       "2192        48665399                  1                       C   \n",
       "2193        48665399                  1                       C   \n",
       "2194        48665399                  1                       C   \n",
       "2195        48665399                  1                       C   \n",
       "2196        48665399                  1                       C   \n",
       "\n",
       "     mutated_from_allele mutated_to_allele         consequence_type  \\\n",
       "0                      C                 A             exon_variant   \n",
       "1                      C                 A         missense_variant   \n",
       "2                      C                 A             exon_variant   \n",
       "3                      C                 A         missense_variant   \n",
       "4                      G                 A              stop_gained   \n",
       "...                  ...               ...                      ...   \n",
       "2192                   C                 T  downstream_gene_variant   \n",
       "2193                   C                 T  downstream_gene_variant   \n",
       "2194                   C                 T  downstream_gene_variant   \n",
       "2195                   C                 T  downstream_gene_variant   \n",
       "2196                   C                 T  downstream_gene_variant   \n",
       "\n",
       "        gene_affected  total_read_count project_code  \n",
       "0     ENSG00000152932                68      BRCA-US  \n",
       "1     ENSG00000152932                68      BRCA-US  \n",
       "2     ENSG00000152932               349      BRCA-US  \n",
       "3     ENSG00000152932               349      BRCA-US  \n",
       "4     ENSG00000185274                53      BRCA-US  \n",
       "...               ...               ...          ...  \n",
       "2192  ENSG00000225697               292      BRCA-US  \n",
       "2193  ENSG00000225697               292      BRCA-US  \n",
       "2194  ENSG00000225697               292      BRCA-US  \n",
       "2195  ENSG00000225697               292      BRCA-US  \n",
       "2196  ENSG00000225697               292      BRCA-US  \n",
       "\n",
       "[2197 rows x 13 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Choice of features here is due to quick look at the columns. Change as you see fit.\n",
    "features = ['icgc_mutation_id', \n",
    "            'icgc_donor_id', \n",
    "            'chromosome', \n",
    "            'chromosome_start',\n",
    "            'chromosome_end',\n",
    "            'chromosome_strand',\n",
    "            'reference_genome_allele',\n",
    "            'mutated_from_allele',\n",
    "            'mutated_to_allele',\n",
    "            'consequence_type',\n",
    "            #'aa_mutation',\n",
    "            #'cds_mutation',\n",
    "            'gene_affected',\n",
    "            #'quality_score',\n",
    "            #'probability',\n",
    "            'total_read_count',\n",
    "            #'seq_coverage', \n",
    "            'project_code']\n",
    "df[features]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Mutation IDs\n",
    "- There are many duplicates. There are only 213 unique mutation IDs for this donor."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 213 unique mutation IDs\n"
     ]
    }
   ],
   "source": [
    "print('There are {} unique mutation IDs'.format(len(set(df[features]['icgc_mutation_id']))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There is only 1 donor in this file\n"
     ]
    }
   ],
   "source": [
    "print('There is only {} donor in this file'.format(len(set(df[features]['icgc_donor_id']))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "data = df[features]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_mutation_id</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_specimen_id</th>\n",
       "      <th>matched_icgc_sample_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>chromosome_end</th>\n",
       "      <th>chromosome_strand</th>\n",
       "      <th>assembly_version</th>\n",
       "      <th>...</th>\n",
       "      <th>probability</th>\n",
       "      <th>total_read_count</th>\n",
       "      <th>mutant_allele_read_count</th>\n",
       "      <th>verification_status</th>\n",
       "      <th>consequence_type</th>\n",
       "      <th>aa_mutation</th>\n",
       "      <th>cds_mutation</th>\n",
       "      <th>gene_affected</th>\n",
       "      <th>transcript_affected</th>\n",
       "      <th>gene_build_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>58021869</td>\n",
       "      <td>1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68</td>\n",
       "      <td>12</td>\n",
       "      <td>not tested</td>\n",
       "      <td>exon_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>ENST00000507977</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60348</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>58021869</td>\n",
       "      <td>1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>68</td>\n",
       "      <td>12</td>\n",
       "      <td>not tested</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A98D</td>\n",
       "      <td>293C&gt;A</td>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>ENST00000282878</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>58021869</td>\n",
       "      <td>1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>349</td>\n",
       "      <td>49</td>\n",
       "      <td>not tested</td>\n",
       "      <td>exon_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>ENST00000507977</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MU22195</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>5</td>\n",
       "      <td>58021869</td>\n",
       "      <td>58021869</td>\n",
       "      <td>1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>349</td>\n",
       "      <td>49</td>\n",
       "      <td>not tested</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A98D</td>\n",
       "      <td>293C&gt;A</td>\n",
       "      <td>ENSG00000152932</td>\n",
       "      <td>ENST00000282878</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MU23549</td>\n",
       "      <td>DO5200</td>\n",
       "      <td>BRCA-US</td>\n",
       "      <td>SP10944</td>\n",
       "      <td>SA60324</td>\n",
       "      <td>7</td>\n",
       "      <td>70886058</td>\n",
       "      <td>70886058</td>\n",
       "      <td>1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>53</td>\n",
       "      <td>37</td>\n",
       "      <td>not tested</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>W310*</td>\n",
       "      <td>929G&gt;A</td>\n",
       "      <td>ENSG00000185274</td>\n",
       "      <td>ENST00000333538</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_mutation_id icgc_donor_id project_code icgc_specimen_id  \\\n",
       "0          MU22195        DO5200      BRCA-US          SP10944   \n",
       "1          MU22195        DO5200      BRCA-US          SP10944   \n",
       "2          MU22195        DO5200      BRCA-US          SP10944   \n",
       "3          MU22195        DO5200      BRCA-US          SP10944   \n",
       "4          MU23549        DO5200      BRCA-US          SP10944   \n",
       "\n",
       "  matched_icgc_sample_id chromosome  chromosome_start  chromosome_end  \\\n",
       "0                SA60348          5          58021869        58021869   \n",
       "1                SA60348          5          58021869        58021869   \n",
       "2                SA60324          5          58021869        58021869   \n",
       "3                SA60324          5          58021869        58021869   \n",
       "4                SA60324          7          70886058        70886058   \n",
       "\n",
       "   chromosome_strand assembly_version  ... probability total_read_count  \\\n",
       "0                  1           GRCh37  ...         NaN               68   \n",
       "1                  1           GRCh37  ...         NaN               68   \n",
       "2                  1           GRCh37  ...         NaN              349   \n",
       "3                  1           GRCh37  ...         NaN              349   \n",
       "4                  1           GRCh37  ...         NaN               53   \n",
       "\n",
       "  mutant_allele_read_count verification_status  consequence_type  aa_mutation  \\\n",
       "0                       12          not tested      exon_variant          NaN   \n",
       "1                       12          not tested  missense_variant         A98D   \n",
       "2                       49          not tested      exon_variant          NaN   \n",
       "3                       49          not tested  missense_variant         A98D   \n",
       "4                       37          not tested       stop_gained        W310*   \n",
       "\n",
       "   cds_mutation    gene_affected transcript_affected gene_build_version  \n",
       "0           NaN  ENSG00000152932     ENST00000507977                 75  \n",
       "1        293C>A  ENSG00000152932     ENST00000282878                 75  \n",
       "2           NaN  ENSG00000152932     ENST00000507977                 75  \n",
       "3        293C>A  ENSG00000152932     ENST00000282878                 75  \n",
       "4        929G>A  ENSG00000185274     ENST00000333538                 75  \n",
       "\n",
       "[5 rows x 25 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Not all mutations are 1bp\n",
    "Some mutations involve longer sequences!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mutation_type</th>\n",
       "      <th>reference_genome_allele</th>\n",
       "      <th>mutated_from_allele</th>\n",
       "      <th>mutated_to_allele</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>159</th>\n",
       "      <td>deletion of &lt;=200bp</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>deletion of &lt;=200bp</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>deletion of &lt;=200bp</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>deletion of &lt;=200bp</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>deletion of &lt;=200bp</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>TGCGGAGATTCTCTTCCTC</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           mutation_type reference_genome_allele  mutated_from_allele  \\\n",
       "159  deletion of <=200bp     TGCGGAGATTCTCTTCCTC  TGCGGAGATTCTCTTCCTC   \n",
       "160  deletion of <=200bp     TGCGGAGATTCTCTTCCTC  TGCGGAGATTCTCTTCCTC   \n",
       "161  deletion of <=200bp     TGCGGAGATTCTCTTCCTC  TGCGGAGATTCTCTTCCTC   \n",
       "162  deletion of <=200bp     TGCGGAGATTCTCTTCCTC  TGCGGAGATTCTCTTCCTC   \n",
       "163  deletion of <=200bp     TGCGGAGATTCTCTTCCTC  TGCGGAGATTCTCTTCCTC   \n",
       "\n",
       "    mutated_to_allele  \n",
       "159                 -  \n",
       "160                 -  \n",
       "161                 -  \n",
       "162                 -  \n",
       "163                 -  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df['mutation_type'] == 'deletion of <=200bp'][['mutation_type', 'reference_genome_allele', 'mutated_from_allele', 'mutated_to_allele']].head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'3_prime_UTR_variant',\n",
       " '5_prime_UTR_premature_start_codon_gain_variant',\n",
       " '5_prime_UTR_variant',\n",
       " 'disruptive_inframe_deletion',\n",
       " 'downstream_gene_variant',\n",
       " 'exon_variant',\n",
       " 'frameshift_variant',\n",
       " 'inframe_deletion',\n",
       " 'intron_variant',\n",
       " 'missense_variant',\n",
       " 'splice_acceptor_variant',\n",
       " 'splice_donor_variant',\n",
       " 'splice_region_variant',\n",
       " 'stop_gained',\n",
       " 'synonymous_variant',\n",
       " 'upstream_gene_variant'}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(df['consequence_type'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
