{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pyensembl import EnsemblRelease"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/lava-gpu/anaconda3/envs/ml-lava/lib/python3.7/site-packages/IPython/core/interactiveshell.py:3063: DtypeWarning: Columns (2) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "gene_encodings = pd.read_csv(\"Merged_Gene_freq_encoding.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Unnamed: 0.1</th>\n",
       "      <th>DonorIDs</th>\n",
       "      <th>CancerType</th>\n",
       "      <th>ENSG00000000003</th>\n",
       "      <th>ENSG00000000005</th>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <th>ENSG00000000457</th>\n",
       "      <th>ENSG00000000460</th>\n",
       "      <th>ENSG00000000938</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000273456</th>\n",
       "      <th>ENSG00000273466</th>\n",
       "      <th>ENSG00000273471</th>\n",
       "      <th>ENSG00000273476</th>\n",
       "      <th>ENSG00000273477</th>\n",
       "      <th>ENSG00000273478</th>\n",
       "      <th>ENSG00000273483</th>\n",
       "      <th>ENSG00000273485</th>\n",
       "      <th>ENSG00000273488</th>\n",
       "      <th>ENSG00000273489</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>DO48566BLCA-US</td>\n",
       "      <td>BLCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>DO223588BLCA-US</td>\n",
       "      <td>BLCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>DO51948BLCA-US</td>\n",
       "      <td>BLCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>DO514BLCA-US</td>\n",
       "      <td>BLCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>DO474BLCA-US</td>\n",
       "      <td>BLCA-US</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5863</th>\n",
       "      <td>2499</td>\n",
       "      <td>5863</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5864</th>\n",
       "      <td>2500</td>\n",
       "      <td>5864</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5865</th>\n",
       "      <td>2501</td>\n",
       "      <td>5865</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5866</th>\n",
       "      <td>2502</td>\n",
       "      <td>5866</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5867</th>\n",
       "      <td>2503</td>\n",
       "      <td>5867</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Healthy</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5868 rows × 30547 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Unnamed: 0  Unnamed: 0.1         DonorIDs CancerType  ENSG00000000003  \\\n",
       "0              0             0   DO48566BLCA-US    BLCA-US              0.0   \n",
       "1              1             1  DO223588BLCA-US    BLCA-US              0.0   \n",
       "2              2             2   DO51948BLCA-US    BLCA-US              0.0   \n",
       "3              3             3     DO514BLCA-US    BLCA-US              0.0   \n",
       "4              4             4     DO474BLCA-US    BLCA-US              0.0   \n",
       "...          ...           ...              ...        ...              ...   \n",
       "5863        2499          5863              NaN    Healthy              0.0   \n",
       "5864        2500          5864              NaN    Healthy              0.0   \n",
       "5865        2501          5865              NaN    Healthy              0.0   \n",
       "5866        2502          5866              NaN    Healthy              0.0   \n",
       "5867        2503          5867              NaN    Healthy              0.0   \n",
       "\n",
       "      ENSG00000000005  ENSG00000000419  ENSG00000000457  ENSG00000000460  \\\n",
       "0                 0.0              0.0              2.0              3.0   \n",
       "1                 0.0              0.0              0.0              1.0   \n",
       "2                 0.0              0.0              0.0              1.0   \n",
       "3                 0.0              0.0              0.0              1.0   \n",
       "4                 0.0              0.0              1.0              1.0   \n",
       "...               ...              ...              ...              ...   \n",
       "5863              0.0              0.0              0.0              0.0   \n",
       "5864              0.0              0.0              0.0              0.0   \n",
       "5865              0.0              0.0              0.0              0.0   \n",
       "5866              0.0              0.0              0.0              0.0   \n",
       "5867              0.0              0.0              0.0              0.0   \n",
       "\n",
       "      ENSG00000000938  ...  ENSG00000273456  ENSG00000273466  ENSG00000273471  \\\n",
       "0                 0.0  ...              0.0              0.0              0.0   \n",
       "1                 0.0  ...              0.0              1.0              0.0   \n",
       "2                 0.0  ...              0.0              0.0              0.0   \n",
       "3                 3.0  ...              0.0              0.0              0.0   \n",
       "4                 0.0  ...              0.0              0.0              0.0   \n",
       "...               ...  ...              ...              ...              ...   \n",
       "5863              0.0  ...              0.0              0.0              0.0   \n",
       "5864              0.0  ...              0.0              0.0              0.0   \n",
       "5865              0.0  ...              0.0              0.0              0.0   \n",
       "5866              0.0  ...              0.0              0.0              0.0   \n",
       "5867              0.0  ...              0.0              0.0              0.0   \n",
       "\n",
       "      ENSG00000273476  ENSG00000273477  ENSG00000273478  ENSG00000273483  \\\n",
       "0                 0.0              0.0              0.0              0.0   \n",
       "1                 0.0              0.0              0.0              1.0   \n",
       "2                 0.0              0.0              0.0              0.0   \n",
       "3                 0.0              0.0              0.0              0.0   \n",
       "4                 0.0              0.0              0.0              0.0   \n",
       "...               ...              ...              ...              ...   \n",
       "5863              0.0              0.0              0.0              0.0   \n",
       "5864              0.0              0.0              0.0              0.0   \n",
       "5865              0.0              0.0              0.0              0.0   \n",
       "5866              0.0              0.0              0.0              0.0   \n",
       "5867              0.0              0.0              0.0              0.0   \n",
       "\n",
       "      ENSG00000273485  ENSG00000273488  ENSG00000273489  \n",
       "0                 0.0              0.0              0.0  \n",
       "1                 0.0              0.0              0.0  \n",
       "2                 0.0              0.0              0.0  \n",
       "3                 0.0              0.0              0.0  \n",
       "4                 0.0              0.0              0.0  \n",
       "...               ...              ...              ...  \n",
       "5863              0.0              0.0              0.0  \n",
       "5864              0.0              0.0              0.0  \n",
       "5865              0.0              0.0              0.0  \n",
       "5866              0.0              0.0              0.0  \n",
       "5867              0.0              0.0              0.0  \n",
       "\n",
       "[5868 rows x 30547 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_encodings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_pathways_csv(filename):\n",
    "    pathway_mappings = {}\n",
    "    with open(filename, \"r\") as csv_file:\n",
    "        for line in csv_file:\n",
    "            stripped = line.replace('\"', '')\n",
    "            tokens = stripped.split(',')\n",
    "            gene_names = [name.strip() for name in tokens[6:]]\n",
    "            pathway_mappings[tokens[0]] = gene_names\n",
    "    return pathway_mappings\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathway_mappings = read_pathways_csv(\"pathways.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1657"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pathway_mappings.items())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "def build_reverse_pathway_lookup(pathway_mappings):\n",
    "    reverse_lookup_table = {}\n",
    "    \n",
    "    for pathway, gene_names in pathway_mappings.items():\n",
    "        for name in gene_names:\n",
    "            try:\n",
    "                reverse_lookup_table[name].append(pathway)\n",
    "            except Exception:\n",
    "                reverse_lookup_table[name] = [pathway]\n",
    "    return reverse_lookup_table\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "reverse_pathway_lookup = build_reverse_pathway_lookup(pathway_mappings)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8522"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(reverse_pathway_lookup.items())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "ensembl_data = EnsemblRelease(77)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene = ensembl_data.gene_by_id('ENSG00000000003')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene(gene_id='ENSG00000000003', gene_name='TSPAN6', biotype='protein_coding', contig='X', start=100627109, end=100639991, strand='-', genome='GRCh38')"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['SUMOylation of transcription cofactors',\n",
       " 'SUMOylation of RNA binding proteins',\n",
       " 'SUMOylation of DNA methylation proteins',\n",
       " 'SUMOylation of DNA damage response and repair proteins',\n",
       " 'SUMOylation of chromatin organization proteins',\n",
       " 'SUMOylation',\n",
       " 'SUMO E3 ligases SUMOylate target proteins',\n",
       " 'Post-translational protein modification',\n",
       " 'Oxidative Stress Induced Senescence',\n",
       " 'Cellular Senescence',\n",
       " 'Cellular responses to stress']"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reverse_pathway_lookup['CBX2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating directory /home/lava-gpu/.config/bioservices \n"
     ]
    }
   ],
   "source": [
    "from bioservices.kegg import KEGG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_ids = [col_name for col_name in gene_encodings.columns if col_name.startswith(\"ENSG\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating directory /home/lava-gpu/.cache/bioservices \n",
      "Welcome to Bioservices\n",
      "======================\n",
      "It looks like you do not have a configuration file.\n",
      "We are creating one with default values in /home/lava-gpu/.config/bioservices/bioservices.cfg .\n",
      "Done\n"
     ]
    }
   ],
   "source": [
    "k = KEGG(verbose=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'\\n'"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k.find(\"pathway\", gene.gene_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'hsa:7105\\tTSPAN6, T245, TM4SF6, TSPAN-6; tetraspanin 6\\n'"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k.find(\"hsa\", \"TSPAN6\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "res = k.get_pathway_by_gene(\"7105\", \"hsa\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([1., 0., 0., 0., 0., 0., 0., 0., 0., 0.])"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "zs = np.zeros(10)\n",
    "zs[0] += 1\n",
    "zs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "def build_gene_name_mappings(gene_ids):\n",
    "    gene_name_mappings = {}\n",
    "    for gene_id in gene_ids:\n",
    "        try:\n",
    "            gene = ensembl_data.gene_by_id(gene_id)\n",
    "            name = gene.gene_name\n",
    "            gene_name_mappings[gene_id] = name\n",
    "        except Exception:\n",
    "            None\n",
    "    return gene_name_mappings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_name_table = build_gene_name_mappings(gene_ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "def build_pathway_encodings(gene_encodings, pathway_lookup, reverse_lookup, gene_name_table):\n",
    "    gene_ids = gene_ids = [col_name for col_name in gene_encodings.columns if col_name.startswith(\"ENSG\")]\n",
    "    gene_data = gene_encodings[gene_ids].to_numpy()\n",
    "    pathway_encodings_table = {}\n",
    "    for key in pathway_lookup.keys():\n",
    "        pathway_encodings_table[key] = np.zeros(gene_data.shape[0])\n",
    "    \n",
    "    for i in range(gene_data.shape[0]):\n",
    "        for j in range(gene_data.shape[1]):\n",
    "            if gene_data[i][j] > 0:\n",
    "                try:\n",
    "                    gene_id = gene_ids[i]\n",
    "                    name = gene_name_table[gene_id]\n",
    "                    pathways = reverse_lookup[name]\n",
    "                    for pathway in pathways:\n",
    "                        pathway_encodings_table[pathway][i] += gene_data[i][j]\n",
    "                except Exception:\n",
    "                    None\n",
    "    pathway_encodings = pd.DataFrame.from_dict(pathway_encodings_table)\n",
    "    pathway_encodings[\"CancerType\"] = gene_encodings[\"CancerType\"]\n",
    "    return pathway_encodings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathway_encodings = build_pathway_encodings(\n",
    "    gene_encodings, pathway_mappings, reverse_pathway_lookup, gene_name_table\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Zinc transporters</th>\n",
       "      <th>Zinc influx into cells by the SLC39 gene family</th>\n",
       "      <th>Zinc efflux and compartmentalization by the SLC30 family</th>\n",
       "      <th>ZBP1(DAI) mediated induction of type I IFNs</th>\n",
       "      <th>YAP1- and WWTR1 (TAZ)-stimulated gene expression</th>\n",
       "      <th>Xenobiotics</th>\n",
       "      <th>XBP1(S) activates chaperone genes</th>\n",
       "      <th>WNT mediated activation of DVL</th>\n",
       "      <th>WNT ligand biogenesis and trafficking</th>\n",
       "      <th>WNT5A-dependent internalization of FZD4</th>\n",
       "      <th>...</th>\n",
       "      <th>Acetylcholine regulates insulin secretion</th>\n",
       "      <th>Acetylcholine Neurotransmitter Release Cycle</th>\n",
       "      <th>Acetylcholine binding and downstream events</th>\n",
       "      <th>ABO blood group biosynthesis</th>\n",
       "      <th>ABC transporters in lipid homeostasis</th>\n",
       "      <th>ABC-family proteins mediated transport</th>\n",
       "      <th>Abacavir transport and metabolism</th>\n",
       "      <th>Abacavir transmembrane transport</th>\n",
       "      <th>Abacavir metabolism</th>\n",
       "      <th>CancerType</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BLCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BLCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BLCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BLCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>BLCA-US</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5863</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5864</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5865</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5866</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5867</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Healthy</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5868 rows × 1658 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Zinc transporters  Zinc influx into cells by the SLC39 gene family  \\\n",
       "0                   0.0                                              0.0   \n",
       "1                   0.0                                              0.0   \n",
       "2                   0.0                                              0.0   \n",
       "3                   0.0                                              0.0   \n",
       "4                   0.0                                              0.0   \n",
       "...                 ...                                              ...   \n",
       "5863                0.0                                              0.0   \n",
       "5864                0.0                                              0.0   \n",
       "5865                0.0                                              0.0   \n",
       "5866                0.0                                              0.0   \n",
       "5867                0.0                                              0.0   \n",
       "\n",
       "      Zinc efflux and compartmentalization by the SLC30 family  \\\n",
       "0                                                   0.0          \n",
       "1                                                   0.0          \n",
       "2                                                   0.0          \n",
       "3                                                   0.0          \n",
       "4                                                   0.0          \n",
       "...                                                 ...          \n",
       "5863                                                0.0          \n",
       "5864                                                0.0          \n",
       "5865                                                0.0          \n",
       "5866                                                0.0          \n",
       "5867                                                0.0          \n",
       "\n",
       "      ZBP1(DAI) mediated induction of type I IFNs  \\\n",
       "0                                             0.0   \n",
       "1                                             0.0   \n",
       "2                                             0.0   \n",
       "3                                             0.0   \n",
       "4                                             0.0   \n",
       "...                                           ...   \n",
       "5863                                          0.0   \n",
       "5864                                          0.0   \n",
       "5865                                          0.0   \n",
       "5866                                          0.0   \n",
       "5867                                          0.0   \n",
       "\n",
       "      YAP1- and WWTR1 (TAZ)-stimulated gene expression  Xenobiotics  \\\n",
       "0                                                  0.0          0.0   \n",
       "1                                                  0.0          0.0   \n",
       "2                                                  0.0          0.0   \n",
       "3                                                  0.0          0.0   \n",
       "4                                                  0.0          0.0   \n",
       "...                                                ...          ...   \n",
       "5863                                               0.0          0.0   \n",
       "5864                                               0.0          0.0   \n",
       "5865                                               0.0          0.0   \n",
       "5866                                               0.0          0.0   \n",
       "5867                                               0.0          0.0   \n",
       "\n",
       "      XBP1(S) activates chaperone genes  WNT mediated activation of DVL  \\\n",
       "0                                   0.0                             0.0   \n",
       "1                                   0.0                             0.0   \n",
       "2                                   0.0                             0.0   \n",
       "3                                   0.0                             0.0   \n",
       "4                                   0.0                             0.0   \n",
       "...                                 ...                             ...   \n",
       "5863                                0.0                             0.0   \n",
       "5864                                0.0                             0.0   \n",
       "5865                                0.0                             0.0   \n",
       "5866                                0.0                             0.0   \n",
       "5867                                0.0                             0.0   \n",
       "\n",
       "      WNT ligand biogenesis and trafficking  \\\n",
       "0                                       0.0   \n",
       "1                                       0.0   \n",
       "2                                       0.0   \n",
       "3                                       0.0   \n",
       "4                                       0.0   \n",
       "...                                     ...   \n",
       "5863                                    0.0   \n",
       "5864                                    0.0   \n",
       "5865                                    0.0   \n",
       "5866                                    0.0   \n",
       "5867                                    0.0   \n",
       "\n",
       "      WNT5A-dependent internalization of FZD4  ...  \\\n",
       "0                                         0.0  ...   \n",
       "1                                         0.0  ...   \n",
       "2                                         0.0  ...   \n",
       "3                                         0.0  ...   \n",
       "4                                         0.0  ...   \n",
       "...                                       ...  ...   \n",
       "5863                                      0.0  ...   \n",
       "5864                                      0.0  ...   \n",
       "5865                                      0.0  ...   \n",
       "5866                                      0.0  ...   \n",
       "5867                                      0.0  ...   \n",
       "\n",
       "      Acetylcholine regulates insulin secretion  \\\n",
       "0                                           0.0   \n",
       "1                                           0.0   \n",
       "2                                           0.0   \n",
       "3                                           0.0   \n",
       "4                                           0.0   \n",
       "...                                         ...   \n",
       "5863                                        0.0   \n",
       "5864                                        0.0   \n",
       "5865                                        0.0   \n",
       "5866                                        0.0   \n",
       "5867                                        0.0   \n",
       "\n",
       "      Acetylcholine Neurotransmitter Release Cycle  \\\n",
       "0                                              0.0   \n",
       "1                                              0.0   \n",
       "2                                              0.0   \n",
       "3                                              0.0   \n",
       "4                                              0.0   \n",
       "...                                            ...   \n",
       "5863                                           0.0   \n",
       "5864                                           0.0   \n",
       "5865                                           0.0   \n",
       "5866                                           0.0   \n",
       "5867                                           0.0   \n",
       "\n",
       "      Acetylcholine binding and downstream events  \\\n",
       "0                                             0.0   \n",
       "1                                             0.0   \n",
       "2                                             0.0   \n",
       "3                                             0.0   \n",
       "4                                             0.0   \n",
       "...                                           ...   \n",
       "5863                                          0.0   \n",
       "5864                                          0.0   \n",
       "5865                                          0.0   \n",
       "5866                                          0.0   \n",
       "5867                                          0.0   \n",
       "\n",
       "      ABO blood group biosynthesis  ABC transporters in lipid homeostasis  \\\n",
       "0                              0.0                                    0.0   \n",
       "1                              0.0                                    0.0   \n",
       "2                              0.0                                    0.0   \n",
       "3                              0.0                                    0.0   \n",
       "4                              0.0                                    0.0   \n",
       "...                            ...                                    ...   \n",
       "5863                           0.0                                    0.0   \n",
       "5864                           0.0                                    0.0   \n",
       "5865                           0.0                                    0.0   \n",
       "5866                           0.0                                    0.0   \n",
       "5867                           0.0                                    0.0   \n",
       "\n",
       "      ABC-family proteins mediated transport  \\\n",
       "0                                        0.0   \n",
       "1                                        0.0   \n",
       "2                                        0.0   \n",
       "3                                        0.0   \n",
       "4                                        0.0   \n",
       "...                                      ...   \n",
       "5863                                     0.0   \n",
       "5864                                     0.0   \n",
       "5865                                     0.0   \n",
       "5866                                     0.0   \n",
       "5867                                     0.0   \n",
       "\n",
       "      Abacavir transport and metabolism  Abacavir transmembrane transport  \\\n",
       "0                                   0.0                               0.0   \n",
       "1                                   0.0                               0.0   \n",
       "2                                   0.0                               0.0   \n",
       "3                                   0.0                               0.0   \n",
       "4                                   0.0                               0.0   \n",
       "...                                 ...                               ...   \n",
       "5863                                0.0                               0.0   \n",
       "5864                                0.0                               0.0   \n",
       "5865                                0.0                               0.0   \n",
       "5866                                0.0                               0.0   \n",
       "5867                                0.0                               0.0   \n",
       "\n",
       "      Abacavir metabolism  CancerType  \n",
       "0                     0.0     BLCA-US  \n",
       "1                     0.0     BLCA-US  \n",
       "2                     0.0     BLCA-US  \n",
       "3                     0.0     BLCA-US  \n",
       "4                     0.0     BLCA-US  \n",
       "...                   ...         ...  \n",
       "5863                  0.0     Healthy  \n",
       "5864                  0.0     Healthy  \n",
       "5865                  0.0     Healthy  \n",
       "5866                  0.0     Healthy  \n",
       "5867                  0.0     Healthy  \n",
       "\n",
       "[5868 rows x 1658 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathway_encodings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Zinc transporters: 1625.0\n",
      "Zinc influx into cells by the SLC39 gene family: 0.0\n",
      "Zinc efflux and compartmentalization by the SLC30 family: 1625.0\n",
      "ZBP1(DAI) mediated induction of type I IFNs: 6667.0\n",
      "YAP1- and WWTR1 (TAZ)-stimulated gene expression: 7065.0\n",
      "Xenobiotics: 6407.0\n",
      "XBP1(S) activates chaperone genes: 19589.0\n",
      "WNT mediated activation of DVL: 8114.0\n",
      "WNT ligand biogenesis and trafficking: 20332.0\n",
      "WNT5A-dependent internalization of FZD4: 2425.0\n",
      "WNT5A-dependent internalization of FZD2: 2022.0\n",
      "Wax biosynthesis: 0.0\n",
      "Wax and plasmalogen biosynthesis: 1627.0\n",
      "VxPx cargo-targeting to cilium: 6450.0\n",
      "Voltage gated Potassium channels: 36026.0\n",
      "VLDLR internalisation and degradation: 2232.0\n",
      "VLDL clearance: 1426.0\n",
      "VLDL assembly: 117.0\n",
      "Vitamins B6 activation to pyridoxal phosphate: 0.0\n",
      "Vitamins: 11994.0\n",
      "Vitamin D (calciferol) metabolism: 5103.0\n",
      "Vitamin C (ascorbate) metabolism: 2250.0\n",
      "Vitamin B5 (pantothenate) metabolism: 5295.0\n",
      "Vitamin B2 (riboflavin) metabolism: 1087.0\n",
      "Vitamin B1 (thiamin) metabolism: 1383.0\n",
      "Visual phototransduction: 39638.0\n",
      "VEGFR2 mediated vascular permeability: 7312.0\n",
      "VEGFR2 mediated cell proliferation: 5872.0\n",
      "VEGF ligand-receptor interactions: 4195.0\n",
      "VEGF binds to VEGFR leading to receptor dimerization: 4195.0\n",
      "VEGFA-VEGFR2 Pathway: 27884.0\n",
      "Vasopressin regulates renal water homeostasis via Aquaporins: 13891.0\n",
      "Vasopressin-like receptors: 10504.0\n",
      "Utilization of Ketone Bodies: 274.0\n",
      "Urea cycle: 3240.0\n",
      "Unwinding of DNA: 3742.0\n",
      "Unfolded Protein Response (UPR): 37049.0\n",
      "Unblocking of NMDA receptors: 3678.0\n",
      "UCH proteinases: 54741.0\n",
      "Ub-specific processing proteases: 63884.0\n",
      "Ubiquitin Mediated Degradation of Phosphorylated Cdc25A: 21732.0\n",
      "Ubiquitin-dependent degradation of Cyclin D: 23465.0\n",
      "Ubiquinol biosynthesis: 3140.0\n",
      "TYSND1 cleaves peroxisomal proteins: 3471.0\n",
      "Type II Na+/Pi cotransporters: 0.0\n",
      "Type I hemidesmosome assembly: 847.0\n",
      "TWIK related potassium channel (TREK): 106.0\n",
      "Tryptophan catabolism: 1655.0\n",
      "TRP channels: 3651.0\n",
      "tRNA processing in the nucleus: 39979.0\n",
      "tRNA processing in the mitochondrion: 503.0\n",
      "tRNA processing: 43161.0\n",
      "tRNA modification in the nucleus and cytosol: 3050.0\n",
      "tRNA modification in the mitochondrion: 132.0\n",
      "tRNA Aminoacylation: 7066.0\n",
      "Tristetraprolin (TTP: 8393.0\n",
      "Triglyceride metabolism: 22031.0\n",
      "Triglyceride catabolism: 17718.0\n",
      "Triglyceride biosynthesis: 4313.0\n",
      "TRIF-mediated programmed cell death: 3858.0\n",
      "TRIF(TICAM1)-mediated TLR4 signaling: 36999.0\n",
      "Transport to the Golgi and subsequent modification: 118765.0\n",
      "Transport of vitamins: 14590.0\n",
      "Transport of the SLBP independent Mature mRNA: 14728.0\n",
      "Transport of the SLBP Dependant Mature mRNA: 14728.0\n",
      "Transport of organic anions: 8690.0\n",
      "Transport of nucleotide sugars: 3097.0\n",
      "Transport of nucleosides and free purine and pyrimidine bases across the plasma membrane: 1175.0\n",
      "Transport of Mature Transcript to Cytoplasm: 42972.0\n",
      "Transport of Mature mRNAs Derived from Intronless Transcripts: 18119.0\n",
      "Transport of Mature mRNA Derived from an Intronless Transcript: 18119.0\n",
      "Transport of Mature mRNA derived from an Intron-Containing Transcript: 39581.0\n",
      "Transport of inorganic cations/anions and amino acids/oligopeptides: 47185.0\n",
      "Transport of gamma-carboxylated protein precursors from the endoplasmic reticulum to the Golgi apparatus: 6647.0\n",
      "Transport of fatty acids: 0.0\n",
      "Transport of bile salts and organic acids: 10622.0\n",
      "Transport and synthesis of PAPS: 0.0\n",
      "Transmission across Electrical Synapses: 1766.0\n",
      "Transmission across Chemical Synapses: 123011.0\n",
      "Translocation of ZAP-70 to Immunological synapse: 2080.0\n",
      "Translocation of SLC2A4 (GLUT4) to the plasma membrane: 22235.0\n",
      "Translesion synthesis by Y family DNA polymerases bypasses lesions on DNA template: 11170.0\n",
      "Translesion synthesis by REV1: 5175.0\n",
      "Translesion synthesis by POLK: 6802.0\n",
      "Translesion synthesis by POLI: 7255.0\n",
      "Translesion Synthesis by POLH: 3309.0\n",
      "Translation initiation complex formation: 60565.0\n",
      "Translation: 148056.0\n",
      "trans-Golgi Network Vesicle Budding: 19500.0\n",
      "Transferrin endocytosis and recycling: 12261.0\n",
      "Transcription of E2F targets under negative control by p107 (RBL1) and p130 (RBL2) in complex with HDAC1: 6787.0\n",
      "Transcription of E2F targets under negative control by DREAM complex: 8249.0\n",
      "Transcription from mitochondrial promoters: 1567.0\n",
      "Transcription-Coupled Nucleotide Excision Repair (TC-NER): 22264.0\n",
      "Transcriptional regulation of white adipocyte differentiation: 85351.0\n",
      "Transcriptional regulation of pluripotent stem cells: 8575.0\n",
      "Transcriptional regulation of granulopoiesis: 15839.0\n",
      "Transcriptional Regulation by VENTX: 26181.0\n",
      "Transcriptional Regulation by TP53: 143636.0\n",
      "Transcriptional regulation by the AP-2 (TFAP2) family of transcription factors: 10501.0\n",
      "Transcriptional regulation by small RNAs: 18977.0\n",
      "Transcriptional regulation by RUNX3: 53219.0\n",
      "Transcriptional regulation by RUNX2: 52531.0\n",
      "Transcriptional regulation by RUNX1: 68369.0\n",
      "Transcriptional Regulation by MECP2: 17455.0\n",
      "Transcriptional Regulation by E2F6: 20008.0\n",
      "Transcriptional activity of SMAD2/SMAD3:SMAD4 heterotrimer: 16220.0\n",
      "Transcriptional activation of p53 responsive genes: 1349.0\n",
      "Transcriptional activation of mitochondrial biogenesis: 19298.0\n",
      "Transcriptional activation of cell cycle inhibitor p21: 1349.0\n",
      "TRAIL signaling: 8454.0\n",
      "Trafficking of myristoylated proteins to the cilium: 3198.0\n",
      "Trafficking of GluR2-containing AMPA receptors: 4888.0\n",
      "Trafficking of AMPA receptors: 5625.0\n",
      "Trafficking and processing of endosomal TLR: 3251.0\n",
      "TRAF6 mediated NF-kB activation: 6973.0\n",
      "TRAF6 mediated IRF7 activation in TLR7/8 or 9 signaling: 1853.0\n",
      "TRAF6 mediated IRF7 activation: 10152.0\n",
      "TRAF6-mediated induction of TAK1 complex within TLR4 complex: 687.0\n",
      "TRAF6 mediated induction of NFkB and MAP kinases upon TLR7/8 or 9 activation: 29979.0\n",
      "TRAF3-dependent IRF activation pathway: 6957.0\n",
      "TP53 regulates transcription of several additional cell death genes whose specific roles in p53-dependent apoptosis remain uncertain: 7320.0\n",
      "TP53 Regulates Transcription of Genes Involved in G2 Cell Cycle Arrest: 5358.0\n",
      "TP53 Regulates Transcription of Genes Involved in G1 Cell Cycle Arrest: 9389.0\n",
      "TP53 Regulates Transcription of Genes Involved in Cytochrome C Release: 21966.0\n",
      "TP53 Regulates Transcription of DNA Repair Genes: 23697.0\n",
      "TP53 Regulates Transcription of Death Receptors and Ligands: 2090.0\n",
      "TP53 Regulates Transcription of Cell Death Genes: 33651.0\n",
      "TP53 Regulates Transcription of Cell Cycle Genes: 22299.0\n",
      "TP53 Regulates Transcription of Caspase Activators and Caspases: 3175.0\n",
      "TP53 regulates transcription of additional cell cycle genes whose exact role in the p53 pathway remain uncertain: 7552.0\n",
      "TP53 Regulates Metabolic Genes: 24683.0\n",
      "Toll Like Receptor TLR6:TLR2 Cascade: 28326.0\n",
      "Toll Like Receptor TLR1:TLR2 Cascade: 28326.0\n",
      "Toll-like Receptor Cascades: 47056.0\n",
      "Toll Like Receptor 9 (TLR9) Cascade: 31600.0\n",
      "Toll Like Receptor 7/8 (TLR7/8) Cascade: 29979.0\n",
      "Toll Like Receptor 5 (TLR5) Cascade: 28126.0\n",
      "Toll Like Receptor 4 (TLR4) Cascade: 42067.0\n",
      "Toll Like Receptor 3 (TLR3) Cascade: 36999.0\n",
      "Toll Like Receptor 2 (TLR2) Cascade: 28326.0\n",
      "Toll Like Receptor 10 (TLR10) Cascade: 28126.0\n",
      "TNF signaling: 13052.0\n",
      "TNFs bind their physiological receptors: 15628.0\n",
      "TNF receptor superfamily (TNFSF) members mediating non-canonical NF-kB pathway: 9035.0\n",
      "TNFR2 non-canonical NF-kB pathway: 44063.0\n",
      "TNFR1-mediated ceramide production: 569.0\n",
      "TNFR1-induced proapoptotic signaling: 6576.0\n",
      "TNFR1-induced NFkappaB signaling pathway: 6499.0\n",
      "TLR3-mediated TICAM1-dependent programmed cell death: 3858.0\n",
      "Tight junction interactions: 2901.0\n",
      "Tie2 Signaling: 13253.0\n",
      "TICAM1-dependent activation of IRF3/IRF7: 1512.0\n",
      "TICAM1: 3163.0\n",
      "Thyroxine biosynthesis: 3364.0\n",
      "Thromboxane signalling through TP receptor: 2219.0\n",
      "Thrombin signalling through proteinase activated receptors (PARs): 3018.0\n",
      "Threonine catabolism: 107.0\n",
      "The role of GTSE1 in G2/M progression after G2 checkpoint: 26932.0\n",
      "The retinoid cycle in cones (daylight vision): 1461.0\n",
      "The proton buffering model: 1659.0\n",
      "The phototransduction cascade: 14861.0\n",
      "The NLRP3 inflammasome: 3652.0\n",
      "The NLRP1 inflammasome: 272.0\n",
      "The fatty acid cycling model: 1659.0\n",
      "The citric acid (TCA) cycle and respiratory electron transport: 53851.0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The canonical retinoid cycle in rods (twilight vision): 8438.0\n",
      "The AIM2 inflammasome: 44.0\n",
      "The activation of arylsulfatases: 3778.0\n",
      "TGF-beta receptor signaling in EMT (epithelial to mesenchymal transition): 18540.0\n",
      "TGF-beta receptor signaling activates SMADs: 28524.0\n",
      "TFAP2A acts as a transcriptional repressor during retinoic acid induced cell differentiation: 1486.0\n",
      "TFAP2 (AP-2) family regulates transcription of other transcription factors: 62.0\n",
      "TFAP2 (AP-2) family regulates transcription of growth factors and their receptors: 1997.0\n",
      "TFAP2 (AP-2) family regulates transcription of cell cycle factors: 3201.0\n",
      "Tetrahydrobiopterin (BH4) synthesis: 2056.0\n",
      "TET1: 0.0\n",
      "Termination of translesion DNA synthesis: 8970.0\n",
      "Termination of O-glycan biosynthesis: 5302.0\n",
      "Terminal pathway of complement: 6469.0\n",
      "Telomere Maintenance: 11130.0\n",
      "Telomere Extension By Telomerase: 0.0\n",
      "Telomere C-strand synthesis initiation: 3971.0\n",
      "Telomere C-strand (Lagging Strand) Synthesis: 10344.0\n",
      "TCR signaling: 36539.0\n",
      "TCF dependent signaling in response to WNT: 68083.0\n",
      "TBC/RABGAPs: 13567.0\n",
      "Tandem pore domain potassium channels: 106.0\n",
      "TAK1 activates NFkB by phosphorylation and activation of IKKs complex: 2362.0\n",
      "Tachykinin receptors bind tachykinins: 2046.0\n",
      "Synthesis of very long-chain fatty acyl-CoAs: 5968.0\n",
      "Synthesis of UDP-N-acetyl-glucosamine: 5206.0\n",
      "Synthesis of substrates in N-glycan biosythesis: 24602.0\n",
      "Synthesis of pyrophosphates in the cytosol: 43.0\n",
      "Synthesis of Prostaglandins (PG) and Thromboxanes (TX): 4891.0\n",
      "Synthesis of PIPs in the nucleus: 0.0\n",
      "Synthesis of PIPs at the plasma membrane: 23330.0\n",
      "Synthesis of PIPs at the late endosome membrane: 6110.0\n",
      "Synthesis of PIPs at the Golgi membrane: 5507.0\n",
      "Synthesis of PIPs at the ER membrane: 258.0\n",
      "Synthesis of PIPs at the early endosome membrane: 7204.0\n",
      "Synthesis of PI: 39.0\n",
      "Synthesis of PG: 6651.0\n",
      "Synthesis of PE: 4049.0\n",
      "Synthesis of PC: 10302.0\n",
      "Synthesis of PA: 11588.0\n",
      "Synthesis of Lipoxins (LX): 1689.0\n",
      "Synthesis of Leukotrienes (LT) and Eoxins (EX): 2101.0\n",
      "Synthesis of Ketone Bodies: 1974.0\n",
      "Synthesis of IPs in the nucleus: 240.0\n",
      "Synthesis of IP3 and IP4 in the cytosol: 9255.0\n",
      "Synthesis of IP2: 5478.0\n",
      "Synthesis of glycosylphosphatidylinositol (GPI): 247.0\n",
      "Synthesis of GDP-mannose: 447.0\n",
      "Synthesis of epoxy (EET) and dihydroxyeicosatrienoic acids (DHET): 1861.0\n",
      "Synthesis of dolichyl-phosphate mannose: 3644.0\n",
      "Synthesis of Dolichyl-phosphate: 1553.0\n",
      "Synthesis of DNA: 87752.0\n",
      "Synthesis of diphthamide-EEF2: 2971.0\n",
      "Synthesis of bile acids and bile salts via 7alpha-hydroxycholesterol: 6452.0\n",
      "Synthesis of bile acids and bile salts via 27-hydroxycholesterol: 1934.0\n",
      "Synthesis of bile acids and bile salts via 24-hydroxycholesterol: 1684.0\n",
      "Synthesis of bile acids and bile salts: 9469.0\n",
      "Synthesis of active ubiquitin: roles of E1 and E2 enzymes: 10318.0\n",
      "Synthesis of 5-eicosatetraenoic acids: 1353.0\n",
      "Synthesis of 15-eicosatetraenoic acid derivatives: 107.0\n",
      "Synthesis of 12-eicosatetraenoic acid derivatives: 1544.0\n",
      "Synthesis of (16-20)-hydroxyeicosatetraenoic acids (HETE): 0.0\n",
      "Synthesis: 7684.0\n",
      "Syndecan interactions: 20788.0\n",
      "Synaptic adhesion-like molecules: 10177.0\n",
      "Switching of origins to a post-replicative state: 77159.0\n",
      "Surfactant metabolism: 3530.0\n",
      "SUMOylation of ubiquitinylation proteins: 15204.0\n",
      "SUMOylation of transcription factors: 2865.0\n",
      "SUMOylation of transcription cofactors: 15789.0\n",
      "SUMOylation of SUMOylation proteins: 14931.0\n",
      "SUMOylation of RNA binding proteins: 18345.0\n",
      "SUMOylation of intracellular receptors: 12546.0\n",
      "SUMOylation of immune response proteins: 2130.0\n",
      "SUMOylation of DNA replication proteins: 15108.0\n",
      "SUMOylation of DNA methylation proteins: 6541.0\n",
      "SUMOylation of DNA damage response and repair proteins: 26890.0\n",
      "SUMOylation of chromatin organization proteins: 22257.0\n",
      "SUMOylation: 58339.0\n",
      "SUMO is transferred from E1 to E2 (UBE2I: 5134.0\n",
      "SUMO is proteolytically processed: 3589.0\n",
      "SUMO is conjugated to E1 (UBA2:SAE1): 3497.0\n",
      "SUMO E3 ligases SUMOylate target proteins: 53555.0\n",
      "Sulfur amino acid metabolism: 12596.0\n",
      "Sulfide oxidation to sulfate: 1078.0\n",
      "Striated Muscle Contraction: 9962.0\n",
      "STING mediated induction of host immune responses: 3359.0\n",
      "Stimuli-sensing channels: 19698.0\n",
      "STAT6-mediated induction of chemokines: 0.0\n",
      "STAT5 Activation: 3046.0\n",
      "Stabilization of p53: 21834.0\n",
      "SRP-dependent cotranslational protein targeting to membrane: 70484.0\n",
      "Spry regulation of FGF signaling: 6272.0\n",
      "Sphingolipid metabolism: 24025.0\n",
      "Sphingolipid de novo biosynthesis: 15163.0\n",
      "S Phase: 111966.0\n",
      "Sperm Motility And Taxes: 144.0\n",
      "SOS-mediated signalling: 1582.0\n",
      "Sodium-coupled sulphate: 0.0\n",
      "Sodium-coupled phosphate cotransporters: 0.0\n",
      "Sodium/Proton exchangers: 2209.0\n",
      "Sodium/Calcium exchangers: 1898.0\n",
      "snRNP Assembly: 22591.0\n",
      "Smooth Muscle Contraction: 15621.0\n",
      "Small interfering RNA (siRNA) biogenesis: 3963.0\n",
      "SMAD2/SMAD3:SMAD4 heterotrimer regulates transcription: 11509.0\n",
      "SMAC(DIABLO)-mediated dissociation of IAP:caspase complexes: 3182.0\n",
      "SMAC (DIABLO) binds to IAPs: 3182.0\n",
      "SMAC: 3182.0\n",
      "SLIT2:ROBO1 increases RHOA activity: 434.0\n",
      "SLC-mediated transmembrane transport: 77591.0\n",
      "SLBP independent Processing of Histone Pre-mRNAs: 3606.0\n",
      "SLBP Dependent Processing of Replication-Dependent Histone Pre-mRNAs: 3606.0\n",
      "SIRT1 negatively regulates rRNA expression: 5629.0\n",
      "Signal transduction by L1: 6296.0\n",
      "Signal regulatory protein family interactions: 7568.0\n",
      "Signalling to STAT3: 0.0\n",
      "Signalling to RAS: 5077.0\n",
      "Signalling to p38 via RIT and RIN: 0.0\n",
      "Signalling to ERKs: 9685.0\n",
      "Signaling by WNT: 96284.0\n",
      "Signaling by VEGF: 32191.0\n",
      "Signaling by Type 1 Insulin-like Growth Factor 1 Receptor (IGF1R): 21982.0\n",
      "Signaling by the B Cell Receptor (BCR): 44762.0\n",
      "Signaling by TGF-beta Receptor Complex: 42147.0\n",
      "Signaling by TGF-beta family members: 56890.0\n",
      "Signaling by SCF-KIT: 12438.0\n",
      "Signaling by ROBO receptors: 116649.0\n",
      "Signaling by Rho GTPases: 159611.0\n",
      "Signaling by Retinoic Acid: 18515.0\n",
      "Signaling by Receptor Tyrosine Kinases: 155254.0\n",
      "Signaling by PTK6: 28199.0\n",
      "Signaling by PDGF: 20894.0\n",
      "Signaling by Nuclear Receptors: 94973.0\n",
      "Signaling by NTRKs: 31965.0\n",
      "Signaling by NTRK3 (TRKC): 5138.0\n",
      "Signaling by NTRK2 (TRKB): 6956.0\n",
      "Signaling by NTRK1 (TRKA): 29864.0\n",
      "Signaling by NOTCH4: 35362.0\n",
      "Signaling by NOTCH3: 17866.0\n",
      "Signaling by NOTCH2: 12888.0\n",
      "Signaling by NOTCH1: 23773.0\n",
      "Signaling by NOTCH: 73561.0\n",
      "Signaling by Non-Receptor Tyrosine Kinases: 28199.0\n",
      "Signaling by NODAL: 6351.0\n",
      "Signaling by MST1: 1350.0\n",
      "Signaling by MET: 19131.0\n",
      "Signaling by Leptin: 3244.0\n",
      "Signaling by Interleukins: 193967.0\n",
      "Signaling by Insulin receptor: 32878.0\n",
      "Signaling by Hippo: 10261.0\n",
      "Signaling by Hedgehog: 43362.0\n",
      "Signaling by GPCR: 272122.0\n",
      "Signaling by FGFR4: 20644.0\n",
      "Signaling by FGFR3: 21222.0\n",
      "Signaling by FGFR2: 30691.0\n",
      "Signaling by FGFR1: 24318.0\n",
      "Signaling by FGFR: 36186.0\n",
      "Signaling by Erythropoietin: 10954.0\n",
      "Signaling by ERBB4: 22313.0\n",
      "Signaling by ERBB2: 22526.0\n",
      "Signaling by EGFR: 22059.0\n",
      "Signaling by BMP: 16509.0\n",
      "Signaling by Activin: 6685.0\n",
      "Signal attenuation: 3001.0\n",
      "Signal amplification: 9112.0\n",
      "Sialic acid metabolism: 13752.0\n",
      "SHC-related events triggered by IGF1R: 1697.0\n",
      "SHC-mediated cascade:FGFR4: 9206.0\n",
      "SHC-mediated cascade:FGFR3: 7839.0\n",
      "SHC-mediated cascade:FGFR2: 9913.0\n",
      "SHC-mediated cascade:FGFR1: 9771.0\n",
      "SHC1 events in ERBB4 signaling: 4432.0\n",
      "SHC1 events in ERBB2 signaling: 6006.0\n",
      "SHC1 events in EGFR signaling: 5953.0\n",
      "Serotonin receptors: 1094.0\n",
      "Serotonin Neurotransmitter Release Cycle: 2434.0\n",
      "Serotonin clearance from the synaptic cleft: 3171.0\n",
      "Serotonin and melatonin biosynthesis: 0.0\n",
      "Serine biosynthesis: 228.0\n",
      "Separation of Sister Chromatids: 62505.0\n",
      "Sensing of DNA Double Strand Breaks: 1667.0\n",
      "Senescence-Associated Secretory Phenotype (SASP): 33926.0\n",
      "Semaphorin interactions: 28507.0\n",
      "Sema4D mediated inhibition of cell attachment and migration: 2602.0\n",
      "Sema4D in semaphorin signaling: 21159.0\n",
      "Sema4D induced cell migration and growth-cone collapse: 18846.0\n",
      "SEMA3A-Plexin repulsion signaling by inhibiting Integrin adhesion: 4227.0\n",
      "Sema3A PAK dependent Axon repulsion: 2795.0\n",
      "Selenocysteine synthesis: 61453.0\n",
      "Selenoamino acid metabolism: 68692.0\n",
      "Selective autophagy: 32933.0\n",
      "SCF-beta-TrCP mediated degradation of Emi1: 42408.0\n",
      "SCF(Skp2)-mediated degradation of p27/p21: 60523.0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scavenging of heme from plasma: 6077.0\n",
      "Scavenging by Class H Receptors: 1750.0\n",
      "Scavenging by Class F Receptors: 1883.0\n",
      "Scavenging by Class B Receptors: 1771.0\n",
      "Scavenging by Class A Receptors: 5009.0\n",
      "RUNX3 regulates YAP1-mediated transcription: 547.0\n",
      "RUNX3 regulates WNT signaling: 1747.0\n",
      "RUNX3 regulates p14-ARF: 17897.0\n",
      "RUNX3 regulates NOTCH signaling: 7495.0\n",
      "RUNX3 Regulates Immune Response and Cell Migration: 2275.0\n",
      "RUNX3 regulates CDKN1A transcription: 17058.0\n",
      "RUNX3 regulates BCL2L11 (BIM) transcription: 1691.0\n",
      "RUNX2 regulates osteoblast differentiation: 8272.0\n",
      "RUNX2 regulates genes involved in differentiation of myeloid cells: 1844.0\n",
      "RUNX2 regulates genes involved in cell migration: 1844.0\n",
      "RUNX2 regulates chondrocyte maturation: 2204.0\n",
      "RUNX2 regulates bone development: 8632.0\n",
      "RUNX1 regulates transcription of genes involved in WNT signaling: 705.0\n",
      "RUNX1 regulates transcription of genes involved in interleukin signaling: 3320.0\n",
      "RUNX1 regulates transcription of genes involved in differentiation of myeloid cells: 2184.0\n",
      "RUNX1 regulates transcription of genes involved in differentiation of keratinocytes: 268.0\n",
      "RUNX1 regulates transcription of genes involved in differentiation of HSCs: 29380.0\n",
      "RUNX1 regulates transcription of genes involved in BCR signaling: 3517.0\n",
      "RUNX1 regulates genes involved in megakaryocyte differentiation and platelet function: 22233.0\n",
      "RUNX1 regulates expression of components of tight junctions: 300.0\n",
      "RUNX1 regulates estrogen receptor mediated transcription: 2323.0\n",
      "RUNX1 interacts with co-factors whose precise effect on RUNX1 targets is not known: 3937.0\n",
      "RUNX1 and FOXP3 control the development of regulatory T lymphocytes (Tregs): 3462.0\n",
      "RSK activation: 2644.0\n",
      "rRNA processing in the nucleus and cytosol: 129755.0\n",
      "rRNA processing in the mitochondrion: 3696.0\n",
      "rRNA processing: 133451.0\n",
      "rRNA modification in the nucleus and cytosol: 47523.0\n",
      "rRNA modification in the mitochondrion: 3320.0\n",
      "ROS and RNS production in phagocytes: 12000.0\n",
      "RORA activates gene expression: 7229.0\n",
      "Role of second messengers in netrin-1 signaling: 3960.0\n",
      "Role of phospholipids in phagocytosis: 7219.0\n",
      "Role of LAT2/NTAL/LAB on calcium mobilization: 5151.0\n",
      "Role of ABL in ROBO-SLIT signaling: 1854.0\n",
      "ROBO receptors bind AKAP5: 3242.0\n",
      "RNA Polymerase I Transcription Termination: 11847.0\n",
      "RNA Polymerase I Transcription Initiation: 18534.0\n",
      "RNA Polymerase I Transcription: 20889.0\n",
      "RNA Polymerase I Promoter Opening: 2366.0\n",
      "RNA Polymerase I Promoter Escape: 11755.0\n",
      "RNA Polymerase I Promoter Clearance: 20889.0\n",
      "RNA Polymerase II Transcription Termination: 32690.0\n",
      "RNA Polymerase II Transcription Pre-Initiation And Promoter Opening: 15717.0\n",
      "RNA Polymerase II Transcription Initiation And Promoter Clearance: 15717.0\n",
      "RNA Polymerase II Transcription Initiation: 15717.0\n",
      "RNA Polymerase II Transcription Elongation: 23827.0\n",
      "RNA Polymerase II Transcription: 439141.0\n",
      "RNA polymerase II transcribes snRNA genes: 22570.0\n",
      "RNA Polymerase II Promoter Escape: 15717.0\n",
      "RNA Polymerase II Pre-transcription Events: 31453.0\n",
      "RNA Polymerase III Transcription Termination: 3559.0\n",
      "RNA Polymerase III Transcription Initiation From Type 3 Promoter: 5357.0\n",
      "RNA Polymerase III Transcription Initiation From Type 2 Promoter: 9847.0\n",
      "RNA Polymerase III Transcription Initiation From Type 1 Promoter: 11341.0\n",
      "RNA Polymerase III Transcription Initiation: 11496.0\n",
      "RNA Polymerase III Transcription: 11687.0\n",
      "RNA Polymerase III Chain Elongation: 3368.0\n",
      "RNA Polymerase III Abortive And Retractive Initiation: 11687.0\n",
      "RNA Pol II CTD phosphorylation and interaction with CE: 11903.0\n",
      "RMTs methylate histone arginines: 8355.0\n",
      "RIP-mediated NFkB activation via ZBP1: 3448.0\n",
      "RIPK1-mediated regulated necrosis: 11849.0\n",
      "Ribosomal scanning and start codon recognition: 60430.0\n",
      "RHO GTPases regulate CFTR trafficking: 3486.0\n",
      "RHO GTPases Activate WASPs and WAVEs: 10175.0\n",
      "RHO GTPases Activate ROCKs: 18939.0\n",
      "RHO GTPases Activate Rhotekin and Rhophilins: 3474.0\n",
      "RHO GTPases activate PKNs: 25391.0\n",
      "RHO GTPases activate PAKs: 19144.0\n",
      "RHO GTPases Activate NADPH Oxidases: 5168.0\n",
      "RHO GTPases activate KTN1: 3690.0\n",
      "RHO GTPases activate IQGAPs: 318.0\n",
      "RHO GTPases Activate Formins: 32137.0\n",
      "RHO GTPases activate CIT: 23206.0\n",
      "RHO GTPase Effectors: 87114.0\n",
      "Rho GTPase cycle: 74564.0\n",
      "Rhesus glycoproteins mediate ammonium transport.: 1542.0\n",
      "Reversible hydration of carbon dioxide: 1615.0\n",
      "Reversal of alkylation damage by DNA dioxygenases: 1927.0\n",
      "RET signaling: 15989.0\n",
      "Retrograde transport at the Trans-Golgi-Network: 32192.0\n",
      "Retrograde neurotrophin signalling: 3319.0\n",
      "Retinoid metabolism and transport: 19651.0\n",
      "Response to metal ions: 3848.0\n",
      "Response to elevated platelet cytosolic Ca2+: 52182.0\n",
      "Response of EIF2AK4 (GCN2) to amino acid deficiency: 67953.0\n",
      "Response of EIF2AK1 (HRI) to heme deficiency: 7164.0\n",
      "Respiratory electron transport: 37590.0\n",
      "Resolution of Sister Chromatid Cohesion: 36355.0\n",
      "Resolution of D-loop Structures through Synthesis-Dependent Strand Annealing (SDSA): 8719.0\n",
      "Resolution of D-loop Structures through Holliday Junction Intermediates: 8719.0\n",
      "Resolution of D-Loop Structures: 8719.0\n",
      "Resolution of AP sites via the single-nucleotide replacement pathway: 707.0\n",
      "Resolution of AP sites via the multiple-nucleotide patch replacement pathway: 6582.0\n",
      "Resolution of Abasic Sites (AP sites): 12755.0\n",
      "Repression of WNT target genes: 3794.0\n",
      "Removal of the Flap Intermediate from the C-strand: 1887.0\n",
      "Removal of the Flap Intermediate: 4082.0\n",
      "Removal of aminoterminal propeptides from gamma-carboxylated proteins: 6647.0\n",
      "Release of Hh-Np from the secreting cell: 0.0\n",
      "Release of apoptotic factors from the mitochondria: 162.0\n",
      "Relaxin receptors: 1442.0\n",
      "Regulation of TP53 Expression and Degradation: 6806.0\n",
      "Regulation of TP53 Degradation: 6748.0\n",
      "Regulation of TP53 Activity through Phosphorylation: 32704.0\n",
      "Regulation of TP53 Activity through Methylation: 84.0\n",
      "Regulation of TP53 Activity through Association with Co-factors: 440.0\n",
      "Regulation of TP53 Activity through Acetylation: 6660.0\n",
      "Regulation of TP53 Activity: 45112.0\n",
      "Regulation of TNFR1 signaling: 12656.0\n",
      "Regulation of TLR by endogenous ligand: 117.0\n",
      "Regulation of the apoptosome activity: 7178.0\n",
      "Regulation of signaling by NODAL: 4776.0\n",
      "Regulation of signaling by CBL: 5352.0\n",
      "Regulation of RUNX3 expression and activity: 38742.0\n",
      "Regulation of RUNX2 expression and activity: 40825.0\n",
      "Regulation of RUNX1 Expression and Activity: 7347.0\n",
      "Regulation of RAS by GAPs: 61616.0\n",
      "Regulation of pyruvate dehydrogenase (PDH) complex: 4666.0\n",
      "Regulation of PTEN stability and activity: 33067.0\n",
      "Regulation of PTEN mRNA translation: 4395.0\n",
      "Regulation of PTEN localization: 0.0\n",
      "Regulation of PTEN gene transcription: 19400.0\n",
      "Regulation of PLK1 Activity at G2/M Transition: 28339.0\n",
      "Regulation of ornithine decarboxylase (ODC): 23658.0\n",
      "Regulation of necroptotic cell death: 11849.0\n",
      "Regulation of mRNA stability by proteins that bind AU-rich elements: 37744.0\n",
      "Regulation of mitotic cell cycle: 49832.0\n",
      "Regulation of MECP2 expression and activity: 11649.0\n",
      "Regulation of localization of FOXO transcription factors: 1575.0\n",
      "Regulation of lipid metabolism by PPARalpha: 93966.0\n",
      "Regulation of KIT signaling: 6307.0\n",
      "Regulation of insulin secretion: 18137.0\n",
      "Regulation of Insulin-like Growth Factor (IGF) transport and uptake by Insulin-like Growth Factor Binding Proteins (IGFBPs): 47513.0\n",
      "Regulation of innate immune responses to cytosolic DNA: 6533.0\n",
      "Regulation of IFNG signaling: 7269.0\n",
      "Regulation of IFNA signaling: 10777.0\n",
      "Regulation of HSF1-mediated heat shock response: 25099.0\n",
      "Regulation of glycolysis by fructose 2: 5170.0\n",
      "Regulation of Glucokinase by Glucokinase Regulatory Protein: 13144.0\n",
      "Regulation of gene expression in late stage (branching morphogenesis) pancreatic bud precursor cells: 5058.0\n",
      "Regulation of gene expression in endocrine-committed (NEUROG3+) progenitor cells: 3056.0\n",
      "Regulation of gene expression in early pancreatic precursor cells: 2037.0\n",
      "Regulation of gene expression in beta cells: 5122.0\n",
      "Regulation of gene expression by Hypoxia-inducible Factor: 7116.0\n",
      "Regulation of gap junction activity: 32.0\n",
      "Regulation of FZD by ubiquitination: 6020.0\n",
      "Regulation of FOXO transcriptional activity by acetylation: 3731.0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Regulation of expression of SLITs and ROBOs: 96945.0\n",
      "Regulation of cytoskeletal remodeling and cell spreading by IPP complex components: 1655.0\n",
      "Regulation of cortical dendrite branching: 155.0\n",
      "Regulation of Complement cascade: 22374.0\n",
      "Regulation of commissural axon pathfinding by SLIT and ROBO: 1655.0\n",
      "Regulation of cholesterol biosynthesis by SREBP (SREBF): 24933.0\n",
      "Regulation of beta-cell development: 12447.0\n",
      "Regulation of Apoptosis: 21885.0\n",
      "Regulation of APC/C activators between G1/S and early anaphase: 48272.0\n",
      "Regulation of activated PAK-2p34 by proteasome mediated degradation: 21732.0\n",
      "Regulation of actin dynamics for phagocytic cup formation: 12106.0\n",
      "Regulation by c-FLIP: 10149.0\n",
      "Regulated proteolysis of p75NTR: 29412.0\n",
      "Regulated Necrosis: 11849.0\n",
      "Reelin signalling pathway: 153.0\n",
      "Reduction of cytosolic Ca++ levels: 2397.0\n",
      "Recycling pathway of L1: 5800.0\n",
      "Recycling of eIF2:GDP: 6113.0\n",
      "Recycling of bile acids and salts: 5394.0\n",
      "Recruitment of NuMA to mitotic centrosomes: 26672.0\n",
      "Recruitment of mitotic centrosome proteins and complexes: 26672.0\n",
      "Recruitment and ATM-mediated phosphorylation of repair and signaling proteins at DNA double strand breaks: 15245.0\n",
      "Recognition of DNA damage by PCNA-containing replication complex: 6968.0\n",
      "Recognition and association of DNA glycosylase with site containing an affected pyrimidine: 7866.0\n",
      "Recognition and association of DNA glycosylase with site containing an affected purine: 4295.0\n",
      "Receptor-type tyrosine-protein phosphatases: 7177.0\n",
      "Receptor Mediated Mitophagy: 8868.0\n",
      "Reactions specific to the complex N-glycan synthesis pathway: 3644.0\n",
      "Ras activation upon Ca2+ influx through NMDA receptor: 2636.0\n",
      "Rap1 signalling: 2770.0\n",
      "RAF-independent MAPK1/3 activation: 1196.0\n",
      "RAF activation: 318.0\n",
      "RAF/MAP kinase cascade: 107114.0\n",
      "Rab regulation of trafficking: 74706.0\n",
      "RA biosynthesis pathway: 12230.0\n",
      "RAB geranylgeranylation: 3244.0\n",
      "RAB GEFs exchange GTP for GDP on RABs: 62725.0\n",
      "Pyruvate metabolism and Citric Acid (TCA) cycle: 16572.0\n",
      "Pyruvate metabolism: 8174.0\n",
      "Pyrimidine salvage: 105.0\n",
      "Pyrimidine catabolism: 105.0\n",
      "Pyrimidine biosynthesis: 2618.0\n",
      "Purine salvage: 1686.0\n",
      "Purine ribonucleoside monophosphate biosynthesis: 0.0\n",
      "Purine catabolism: 1900.0\n",
      "PTK6 Regulates RTKs and Their Effectors AKT1 and DOK1: 10283.0\n",
      "PTK6 Regulates RHO GTPases: 7157.0\n",
      "PTK6 Regulates Proteins Involved in RNA Processing: 11576.0\n",
      "PTK6 Regulates Cell Cycle: 9886.0\n",
      "PTK6 promotes HIF1A stabilization: 8302.0\n",
      "PTK6 Expression: 10186.0\n",
      "PTK6 Down-Regulation: 8362.0\n",
      "PTK6 Activates STAT3: 6735.0\n",
      "PTEN Regulation: 56862.0\n",
      "Proton-coupled monocarboxylate transport: 1524.0\n",
      "Proton/oligopeptide cotransporters: 1570.0\n",
      "Protein ubiquitination: 13837.0\n",
      "Protein repair: 0.0\n",
      "Protein-protein interactions at synapses: 24201.0\n",
      "Protein methylation: 8744.0\n",
      "Protein folding: 58362.0\n",
      "Prostanoid ligand receptors: 3322.0\n",
      "Prostacyclin signalling through prostacyclin receptor: 224.0\n",
      "Propionyl-CoA catabolism: 1848.0\n",
      "Prolonged ERK activation events: 4608.0\n",
      "Proline catabolism: 116.0\n",
      "Prolactin receptor signaling: 6330.0\n",
      "Progressive trimming of alpha-1: 3313.0\n",
      "Processive synthesis on the lagging strand: 6931.0\n",
      "Processive synthesis on the C-strand of the telomere: 4736.0\n",
      "Processing of SMDT1: 2548.0\n",
      "Processing of Intronless Pre-mRNAs: 7615.0\n",
      "Processing of DNA double-strand break ends: 24801.0\n",
      "Processing of Capped Intronless Pre-mRNA: 9370.0\n",
      "Processing of Capped Intron-Containing Pre-mRNA: 129522.0\n",
      "Processing and activation of SUMO: 6780.0\n",
      "Presynaptic phase of homologous DNA pairing and strand exchange: 10350.0\n",
      "Presynaptic nicotinic acetylcholine receptors: 11395.0\n",
      "Presynaptic function of Kainate receptors: 164.0\n",
      "Presynaptic depolarization and calcium channel opening: 89.0\n",
      "Pre-NOTCH Transcription and Translation: 14980.0\n",
      "Pre-NOTCH Processing in the Endoplasmic Reticulum: 3617.0\n",
      "Pre-NOTCH Processing in Golgi: 1142.0\n",
      "Pre-NOTCH Expression and Processing: 19523.0\n",
      "Pregnenolone biosynthesis: 328.0\n",
      "Prefoldin mediated transfer of substrate to CCT/TriC: 14786.0\n",
      "PRC2 methylates histones and DNA: 6475.0\n",
      "PPARA activates gene expression: 93966.0\n",
      "PP2A-mediated dephosphorylation of key metabolic factors: 1791.0\n",
      "POU5F1 (OCT4): 3150.0\n",
      "Potassium transport channels: 0.0\n",
      "Potassium Channels: 59705.0\n",
      "Post-translational protein phosphorylation: 41779.0\n",
      "Post-translational protein modification: 502748.0\n",
      "Post-translational modification: synthesis of GPI-anchored proteins: 36012.0\n",
      "Post-transcriptional silencing by small RNAs: 4395.0\n",
      "Postsynaptic nicotinic acetylcholine receptors: 11395.0\n",
      "Post NMDA receptor activation events: 19929.0\n",
      "Post-chaperonin tubulin folding pathway: 11504.0\n",
      "Positive epigenetic regulation of rRNA expression: 20672.0\n",
      "Polymerase switching on the C-strand of the telomere: 4214.0\n",
      "Polymerase switching: 4214.0\n",
      "Polo-like kinase mediated events: 2958.0\n",
      "POLB-Dependent Long Patch Base Excision Repair: 3058.0\n",
      "PLC-gamma1 signalling: 1868.0\n",
      "PLCG1 events in ERBB2 signaling: 1868.0\n",
      "PLC beta mediated events: 27199.0\n",
      "Platelet sensitization by LDL: 7589.0\n",
      "Platelet homeostasis: 46956.0\n",
      "Platelet degranulation: 49039.0\n",
      "Platelet calcium homeostasis: 32106.0\n",
      "Platelet Aggregation (Plug Formation): 9736.0\n",
      "Platelet Adhesion to exposed collagen: 1677.0\n",
      "Platelet activation: 79079.0\n",
      "Plasmalogen biosynthesis: 1627.0\n",
      "Plasma lipoprotein remodeling: 10746.0\n",
      "Plasma lipoprotein clearance: 11580.0\n",
      "Plasma lipoprotein assembly: 4532.0\n",
      "PKMTs methylate histone lysines: 13175.0\n",
      "PKA-mediated phosphorylation of key metabolic factors: 301.0\n",
      "PKA-mediated phosphorylation of CREB: 6975.0\n",
      "PKA activation in glucagon signalling: 5443.0\n",
      "PKA activation: 5383.0\n",
      "PIWI-interacting RNA (piRNA) biogenesis: 6218.0\n",
      "PIP3 activates AKT signaling: 105463.0\n",
      "Pink/Parkin Mediated Mitophagy: 4706.0\n",
      "PI Metabolism: 29292.0\n",
      "PI5P Regulates TP53 Acetylation: 1659.0\n",
      "PI5P: 32791.0\n",
      "PI3K events in ERBB4 signaling: 4474.0\n",
      "PI3K events in ERBB2 signaling: 7722.0\n",
      "PI-3K cascade:FGFR4: 10922.0\n",
      "PI-3K cascade:FGFR3: 9555.0\n",
      "PI-3K cascade:FGFR2: 11629.0\n",
      "PI-3K cascade:FGFR1: 11487.0\n",
      "PI3K Cascade: 20285.0\n",
      "PI3K/AKT activation: 3540.0\n",
      "Physiological factors: 5179.0\n",
      "Phosphorylation of the APC/C: 4418.0\n",
      "Phosphorylation of proteins involved in the G2/M transition by Cyclin A:Cdc2 complexes: 0.0\n",
      "Phosphorylation of proteins involved in G1/S transition by active Cyclin E:Cdk2 complexes: 26432.0\n",
      "Phosphorylation of Emi1: 19147.0\n",
      "Phosphorylation of CD3 and TCR zeta chains: 380.0\n",
      "Phospholipid metabolism: 67862.0\n",
      "Phospholipase C-mediated cascade: FGFR1: 10057.0\n",
      "Phospholipase C-mediated cascade; FGFR4: 9492.0\n",
      "Phospholipase C-mediated cascade; FGFR3: 8125.0\n",
      "Phospholipase C-mediated cascade; FGFR2: 10199.0\n",
      "Phosphate bond hydrolysis by NUDT proteins: 255.0\n",
      "Phosphate bond hydrolysis by NTPDase proteins: 0.0\n",
      "Phenylalanine metabolism: 0.0\n",
      "Phenylalanine and tyrosine metabolism: 0.0\n",
      "Phase II - Conjugation of compounds: 19348.0\n",
      "Phase I - Functionalization of compounds: 34943.0\n",
      "Phase 4 - resting membrane potential: 1583.0\n",
      "Phase 3 - rapid repolarisation: 127.0\n",
      "Phase 2 - plateau phase: 6024.0\n",
      "Phase 1 - inactivation of fast Na+ channels: 4350.0\n",
      "Phase 0 - rapid depolarisation: 11469.0\n",
      "Peroxisomal protein import: 38347.0\n",
      "Peroxisomal lipid metabolism: 15079.0\n",
      "PERK regulates gene expression: 15601.0\n",
      "Peptide ligand-binding receptors: 83984.0\n",
      "Peptide hormone metabolism: 36787.0\n",
      "Peptide hormone biosynthesis: 5505.0\n",
      "Peptide chain elongation: 60099.0\n",
      "Pentose phosphate pathway: 0.0\n",
      "PECAM1 interactions: 3555.0\n",
      "PD-1 signaling: 3454.0\n",
      "PCP/CE pathway: 30262.0\n",
      "PCNA-Dependent Long Patch Base Excision Repair: 6582.0\n",
      "Passive transport by Aquaporins: 70.0\n",
      "Packaging Of Telomere Ends: 786.0\n",
      "p75NTR signals via NF-kB: 27583.0\n",
      "p75NTR regulates axonogenesis: 32820.0\n",
      "p75NTR recruits signalling complexes: 25889.0\n",
      "p75 NTR receptor-mediated signalling: 62702.0\n",
      "p75NTR negatively regulates cell cycle via SC1: 29345.0\n",
      "p53-Independent G1/S DNA damage checkpoint: 21732.0\n",
      "p53-Independent DNA Damage Response: 21732.0\n",
      "p53-Dependent G1 DNA Damage Response: 26149.0\n",
      "p53-Dependent G1/S DNA damage checkpoint: 26149.0\n",
      "p38MAPK events: 3495.0\n",
      "P2Y receptors: 158.0\n",
      "p130Cas linkage to MAPK signaling for integrins: 5882.0\n",
      "Oxygen-dependent proline hydroxylation of Hypoxia-inducible Factor Alpha: 28339.0\n",
      "Oxidative Stress Induced Senescence: 26798.0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Ovarian tumor domain proteases: 8089.0\n",
      "Other semaphorin interactions: 4606.0\n",
      "Other interleukin signaling: 10301.0\n",
      "Organic cation transport: 1905.0\n",
      "Organic cation/anion/zwitterion transport: 1905.0\n",
      "Organic anion transporters: 3052.0\n",
      "Organic anion transport: 0.0\n",
      "Orexin and neuropeptides FF and QRFP bind to their respective receptors: 1682.0\n",
      "Orc1 removal from chromatin: 31668.0\n",
      "Opsins: 3442.0\n",
      "Opioid Signalling: 43089.0\n",
      "Oncogene Induced Senescence: 16030.0\n",
      "O-linked glycosylation of mucins: 11947.0\n",
      "O-linked glycosylation: 13635.0\n",
      "Oligomerization of connexins into connexons: 0.0\n",
      "Olfactory Signaling Pathway: 3671.0\n",
      "O-glycosylation of TSR domain-containing proteins: 1438.0\n",
      "OAS antiviral response: 4841.0\n",
      "O2/CO2 exchange in erythrocytes: 2405.0\n",
      "Nucleotide salvage: 1791.0\n",
      "Nucleotide-like (purinergic) receptors: 158.0\n",
      "Nucleotide Excision Repair: 37896.0\n",
      "Nucleotide-binding domain: 12013.0\n",
      "Nucleosome assembly: 10810.0\n",
      "Nucleobase catabolism: 5485.0\n",
      "Nucleobase biosynthesis: 2618.0\n",
      "Nuclear signaling by ERBB4: 13405.0\n",
      "Nuclear Receptor transcription pathway: 22466.0\n",
      "Nuclear Pore Complex (NPC) Disassembly: 15719.0\n",
      "Nuclear Events (kinase and transcription factor activation): 13870.0\n",
      "Nuclear Envelope Reassembly: 4834.0\n",
      "Nuclear Envelope Breakdown: 17147.0\n",
      "NTRK3 as a dependence receptor: 64.0\n",
      "NTRK2 activates RAC1: 208.0\n",
      "NRIF signals cell death from the nucleus: 29194.0\n",
      "NrCAM interactions: 2209.0\n",
      "NRAGE signals death through JNK: 48561.0\n",
      "NR1H3 & NR1H2 regulate gene expression linked to cholesterol transport and efflux: 9549.0\n",
      "NR1H2 and NR1H3-mediated signaling: 11900.0\n",
      "NR1H2 & NR1H3 regulate gene expression to limit cholesterol uptake: 326.0\n",
      "NR1H2 & NR1H3 regulate gene expression to control bile acid homeostasis: 229.0\n",
      "NR1H2 & NR1H3 regulate gene expression linked to triglyceride lipolysis in adipose: 114.0\n",
      "NR1H2 & NR1H3 regulate gene expression linked to lipogenesis: 397.0\n",
      "NR1H2 & NR1H3 regulate gene expression linked to gluconeogenesis: 1970.0\n",
      "NR1D1 (REV-ERBA) represses gene expression: 1587.0\n",
      "Notch-HLH transcription pathway: 7565.0\n",
      "NOTCH4 Intracellular Domain Regulates Transcription: 5315.0\n",
      "NOTCH4 Activation and Transmission of Signal to the Nucleus: 5651.0\n",
      "NOTCH3 Intracellular Domain Regulates Transcription: 9394.0\n",
      "NOTCH3 Activation and Transmission of Signal to the Nucleus: 8580.0\n",
      "NOTCH2 intracellular domain regulates transcription: 3255.0\n",
      "NOTCH2 Activation and Transmission of Signal to the Nucleus: 9633.0\n",
      "NOTCH1 Intracellular Domain Regulates Transcription: 15539.0\n",
      "NOSTRIN mediated eNOS trafficking: 1068.0\n",
      "Norepinephrine Neurotransmitter Release Cycle: 3903.0\n",
      "NoRC negatively regulates rRNA expression: 15443.0\n",
      "Nonsense Mediated Decay (NMD) independent of the Exon Junction Complex (EJC): 65947.0\n",
      "Nonsense Mediated Decay (NMD) enhanced by the Exon Junction Complex (EJC): 76649.0\n",
      "Nonsense-Mediated Decay (NMD): 76649.0\n",
      "Non-integrin membrane-ECM interactions: 26733.0\n",
      "Nonhomologous End-Joining (NHEJ): 12113.0\n",
      "Noncanonical activation of NOTCH3: 2401.0\n",
      "NOD1/2 Signaling Pathway: 7757.0\n",
      "Nitric oxide stimulates guanylate cyclase: 7037.0\n",
      "NIK-->noncanonical NF-kB signaling: 23717.0\n",
      "Nicotinate metabolism: 4916.0\n",
      "Nicotinamide salvaging: 2811.0\n",
      "N-glycan trimming in the ER and Calnexin/Calreticulin cycle: 4607.0\n",
      "N-glycan trimming and elongation in the cis-Golgi: 3313.0\n",
      "N-glycan antennae elongation in the medial/trans-Golgi: 10191.0\n",
      "N-Glycan antennae elongation: 6547.0\n",
      "NGF processing: 155.0\n",
      "NGF-independant TRKA activation: 157.0\n",
      "NF-kB is activated and signals survival: 27556.0\n",
      "NF-kB activation through FADD/RIP-1 pathway mediated by caspase-8 and -10: 9525.0\n",
      "NFG and proNGF binds to p75NTR: 25836.0\n",
      "Neutrophil degranulation: 211013.0\n",
      "Neurotransmitter release cycle: 57417.0\n",
      "Neurotransmitter receptors and postsynaptic signal transmission: 63498.0\n",
      "Neurotransmitter clearance: 4914.0\n",
      "Neurophilin interactions with VEGF and VEGFR: 2710.0\n",
      "Neurofascin interactions: 2085.0\n",
      "Neurexins and neuroligins: 13843.0\n",
      "Netrin mediated repulsion signals: 4560.0\n",
      "Netrin-1 signaling: 11712.0\n",
      "Nephrin family interactions: 7376.0\n",
      "Negative regulators of DDX58/IFIH1 signaling: 12863.0\n",
      "Negative regulation of the PI3K/AKT network: 37695.0\n",
      "Negative regulation of TCF-dependent signaling by WNT ligand antagonists: 3620.0\n",
      "Negative regulation of TCF-dependent signaling by DVL-interacting proteins: 532.0\n",
      "Negative regulation of NOTCH4 signaling: 24396.0\n",
      "Negative regulation of NMDA receptor-mediated neuronal transmission: 831.0\n",
      "Negative regulation of MET activity: 5781.0\n",
      "Negative regulation of MAPK pathway: 1544.0\n",
      "Negative regulation of FGFR4 signaling: 13896.0\n",
      "Negative regulation of FGFR3 signaling: 12529.0\n",
      "Negative regulation of FGFR2 signaling: 14603.0\n",
      "Negative regulation of FGFR1 signaling: 14461.0\n",
      "Negative regulation of activity of TFAP2 (AP-2) family transcription factors: 2248.0\n",
      "Negative feedback regulation of MAPK pathway: 1037.0\n",
      "Negative epigenetic regulation of rRNA expression: 17387.0\n",
      "Neddylation: 90199.0\n",
      "Nectin/Necl trans heterodimerization: 119.0\n",
      "NCAM signaling for neurite out-growth: 14116.0\n",
      "NCAM1 interactions: 7696.0\n",
      "NADE modulates death signalling: 27537.0\n",
      "Na+/Cl- dependent neurotransmitter transporters: 1510.0\n",
      "Myogenesis: 10197.0\n",
      "MyD88-independent TLR4 cascade: 36999.0\n",
      "MyD88 dependent cascade initiated on endosome: 29979.0\n",
      "MyD88 cascade initiated on plasma membrane: 28126.0\n",
      "MyD88:MAL(TIRAP) cascade initiated on plasma membrane: 28326.0\n",
      "Muscarinic acetylcholine receptors: 0.0\n",
      "Multifunctional anion exchangers: 618.0\n",
      "mTOR signalling: 30014.0\n",
      "mTORC1-mediated signalling: 23667.0\n",
      "MTF1 activates gene expression: 109.0\n",
      "mRNA Splicing - Minor Pathway: 30379.0\n",
      "mRNA Splicing - Major Pathway: 111122.0\n",
      "mRNA Splicing: 111234.0\n",
      "mRNA Editing: 1548.0\n",
      "mRNA decay by 5' to 3' exoribonuclease: 3323.0\n",
      "mRNA decay by 3' to 5' exoribonuclease: 9863.0\n",
      "mRNA Capping: 13754.0\n",
      "mRNA 3'-end processing: 30935.0\n",
      "M Phase: 117131.0\n",
      "Molybdenum cofactor biosynthesis: 1898.0\n",
      "Molecules associated with elastic fibres: 29760.0\n",
      "Mitotic Telophase/Cytokinesis: 8179.0\n",
      "Mitotic Spindle Checkpoint: 34112.0\n",
      "Mitotic Prophase: 26611.0\n",
      "Mitotic Prometaphase: 70832.0\n",
      "Mitotic Metaphase and Anaphase: 68914.0\n",
      "Mitotic G2-G2/M phases: 68557.0\n",
      "Mitotic G1 phase and G1/S transition: 91509.0\n",
      "Mitotic Anaphase: 67339.0\n",
      "Mitophagy: 10649.0\n",
      "Mitochondrial Uncoupling Proteins: 1659.0\n",
      "Mitochondrial tRNA aminoacylation: 3819.0\n",
      "Mitochondrial translation termination: 39286.0\n",
      "Mitochondrial translation initiation: 39034.0\n",
      "Mitochondrial translation elongation: 39264.0\n",
      "Mitochondrial translation: 42246.0\n",
      "Mitochondrial transcription initiation: 1567.0\n",
      "Mitochondrial protein import: 29937.0\n",
      "Mitochondrial iron-sulfur cluster biogenesis: 94.0\n",
      "mitochondrial fatty acid beta-oxidation of unsaturated fatty acids: 3636.0\n",
      "mitochondrial fatty acid beta-oxidation of saturated fatty acids: 4031.0\n",
      "Mitochondrial Fatty Acid Beta-Oxidation: 11555.0\n",
      "Mitochondrial calcium ion transport: 4677.0\n",
      "Mitochondrial biogenesis: 26524.0\n",
      "Mismatch repair (MMR) directed by MSH2:MSH6 (MutSalpha): 8479.0\n",
      "Mismatch repair (MMR) directed by MSH2:MSH3 (MutSbeta): 8352.0\n",
      "Mismatch Repair: 10286.0\n",
      "Miscellaneous transport and binding events: 1643.0\n",
      "Mineralocorticoid biosynthesis: 22.0\n",
      "MicroRNA (miRNA) biogenesis: 8807.0\n",
      "MHC class II antigen presentation: 19134.0\n",
      "MET receptor recycling: 4110.0\n",
      "MET Receptor Activation: 3747.0\n",
      "MET promotes cell motility: 6361.0\n",
      "MET interacts with TNS proteins: 811.0\n",
      "Methylation: 6499.0\n",
      "Methionine salvage pathway: 1796.0\n",
      "Metal sequestration by antimicrobial proteins: 139.0\n",
      "Metallothioneins bind metals: 3739.0\n",
      "Metalloprotease DUBs: 9883.0\n",
      "Metal ion SLC transporters: 3471.0\n",
      "MET activates STAT3: 811.0\n",
      "MET activates RAS signaling: 2562.0\n",
      "MET activates RAP1 and RAC1: 6056.0\n",
      "MET activates PTPN11: 2485.0\n",
      "MET activates PTK2 signaling: 1116.0\n",
      "MET activates PI3K/AKT signaling: 4109.0\n",
      "Metabolism of water-soluble vitamins and cofactors: 27643.0\n",
      "Metabolism of vitamins and cofactors: 52490.0\n",
      "Metabolism of vitamin K: 0.0\n",
      "Metabolism of steroids: 53467.0\n",
      "Metabolism of steroid hormones: 5544.0\n",
      "Metabolism of porphyrins: 5758.0\n",
      "Metabolism of polyamines: 28754.0\n",
      "Metabolism of nucleotides: 11296.0\n",
      "Metabolism of non-coding RNA: 22591.0\n",
      "Metabolism of nitric oxide: eNOS activation and regulation: 4770.0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Metabolism of lipids: 282070.0\n",
      "Metabolism of ingested SeMet: 5655.0\n",
      "Metabolism of ingested H2SeO4 and H2SeO3 into H2Se: 0.0\n",
      "Metabolism of folate and pterines: 2134.0\n",
      "Metabolism of fat-soluble vitamins: 19651.0\n",
      "Metabolism of cofactors: 5196.0\n",
      "Metabolism of carbohydrates: 97571.0\n",
      "Metabolism of Angiotensinogen to Angiotensins: 11482.0\n",
      "Metabolism of amino acids and derivatives: 135709.0\n",
      "Metabolism of amine-derived hormones: 5183.0\n",
      "Membrane Trafficking: 294047.0\n",
      "Melanin biosynthesis: 1943.0\n",
      "Meiotic synapsis: 9948.0\n",
      "Meiotic recombination: 11817.0\n",
      "Meiosis: 21626.0\n",
      "MECP2 regulates transcription of neuronal ligands: 3415.0\n",
      "MECP2 regulates transcription of genes involved in GABA signaling: 0.0\n",
      "MECP2 regulates transcription factors: 1873.0\n",
      "MECP2 regulates neuronal receptors and channels: 7435.0\n",
      "MASTL Facilitates Mitotic Progression: 1514.0\n",
      "MAPK targets/ Nuclear events mediated by MAP kinases: 18326.0\n",
      "MAP kinase activation: 27843.0\n",
      "MAPK family signaling cascades: 117829.0\n",
      "MAPK6/MAPK4 signaling: 32447.0\n",
      "MAPK3 (ERK1) activation: 1079.0\n",
      "MAPK1 (ERK2) activation: 276.0\n",
      "MAPK1/MAPK3 signaling: 107114.0\n",
      "MAP3K8 (TPL2)-dependent MAPK1/3 activation: 5480.0\n",
      "MAP2K and MAPK activation: 6658.0\n",
      "Major pathway of rRNA processing in the nucleolus and cytosol: 126349.0\n",
      "Macroautophagy: 50139.0\n",
      "Lysosphingolipid and LPA receptors: 4677.0\n",
      "Lysosome Vesicle Biogenesis: 6921.0\n",
      "Lysosomal oligosaccharide catabolism: 1356.0\n",
      "Lysine catabolism: 2868.0\n",
      "LRR FLII-interacting protein 1 (LRRFIP1) activates type I IFN production: 3345.0\n",
      "Loss of proteins required for interphase microtubule organization from the centrosome: 25001.0\n",
      "Loss of Nlp from mitotic centrosomes: 25001.0\n",
      "Long-term potentiation: 2056.0\n",
      "Localization of the PINCH-ILK-PARVIN complex to focal adhesions: 51.0\n",
      "Lipophagy: 17924.0\n",
      "Lipid particle organization: 0.0\n",
      "Linoleic acid (LA) metabolism: 3645.0\n",
      "Ligand-receptor interactions: 27.0\n",
      "LGI-ADAM interactions: 4517.0\n",
      "Lewis blood group biosynthesis: 8998.0\n",
      "Leukotriene receptors: 0.0\n",
      "Lectin pathway of complement activation: 1873.0\n",
      "Leading Strand Synthesis: 4214.0\n",
      "LDL remodeling: 180.0\n",
      "LDL clearance: 6389.0\n",
      "Late endosomal microautophagy: 23176.0\n",
      "Laminin interactions: 6707.0\n",
      "Lagging Strand Synthesis: 8568.0\n",
      "Lactose synthesis: 1826.0\n",
      "L1CAM interactions: 23482.0\n",
      "L13a-mediated translational silencing of Ceruloplasmin expression: 80346.0\n",
      "KSRP (KHSRP) binds and destabilizes mRNA: 8278.0\n",
      "Kinesins: 7909.0\n",
      "Ketone body metabolism: 2152.0\n",
      "Keratinization: 41961.0\n",
      "Keratan sulfate degradation: 3588.0\n",
      "Keratan sulfate biosynthesis: 17422.0\n",
      "Keratan sulfate/keratin metabolism: 17491.0\n",
      "Josephin domain DUBs: 3842.0\n",
      "JNK (c-Jun kinases) phosphorylation and activation mediated by activated human TAK1: 4274.0\n",
      "ISG15 antiviral mechanism: 27526.0\n",
      "IRS-related events triggered by IGF1R: 21982.0\n",
      "IRS-mediated signalling: 21867.0\n",
      "IRS activation: 499.0\n",
      "Iron uptake and transport: 23874.0\n",
      "IRF3-mediated induction of type I IFN: 3359.0\n",
      "IRF3 mediated activation of type 1 IFN: 4886.0\n",
      "IRE1alpha activates chaperones: 19679.0\n",
      "IRAK2 mediated activation of TAK1 complex upon TLR7/8 or 9 stimulation: 135.0\n",
      "IRAK2 mediated activation of TAK1 complex: 135.0\n",
      "IRAK1 recruits IKK complex upon TLR7/8 or 9 stimulation: 26.0\n",
      "IRAK1 recruits IKK complex: 26.0\n",
      "Ion transport by P-type ATPases: 19651.0\n",
      "Ionotropic activity of kainate receptors: 1390.0\n",
      "Ion influx/efflux at host-pathogen interface: 1745.0\n",
      "Ion homeostasis: 25783.0\n",
      "Ion channel transport: 48841.0\n",
      "Inwardly rectifying K+ channels: 21736.0\n",
      "Invadopodia formation: 1512.0\n",
      "Intrinsic Pathway of Fibrin Clot Formation: 11598.0\n",
      "Intrinsic Pathway for Apoptosis: 41484.0\n",
      "Intra-Golgi traffic: 17945.0\n",
      "Intra-Golgi and retrograde Golgi-to-ER traffic: 79563.0\n",
      "Intraflagellar transport: 23517.0\n",
      "Intracellular signaling by second messengers: 121972.0\n",
      "Intracellular oxygen transport: 0.0\n",
      "Intracellular metabolism of fatty acids regulates insulin secretion: 2177.0\n",
      "Intestinal hexose absorption: 96.0\n",
      "Intestinal absorption: 237.0\n",
      "Interleukin receptor SHC signaling: 10362.0\n",
      "Interleukin-9 signaling: 6909.0\n",
      "Interleukin-7 signaling: 6536.0\n",
      "Interleukin-6 signaling: 3769.0\n",
      "Interleukin-6 family signaling: 7403.0\n",
      "Interleukin-4 and Interleukin-13 signaling: 69121.0\n",
      "Interleukin-38 signaling: 3193.0\n",
      "Interleukin-37 signaling: 6717.0\n",
      "Interleukin-36 pathway: 1753.0\n",
      "Interleukin-35 Signalling: 2300.0\n",
      "Interleukin-33 signaling: 1754.0\n",
      "Interleukin-3: 15552.0\n",
      "Interleukin-2 signaling: 6256.0\n",
      "Interleukin-2 family signaling: 17086.0\n",
      "Interleukin-27 signaling: 6583.0\n",
      "Interleukin-23 signaling: 7453.0\n",
      "Interleukin-21 signaling: 3714.0\n",
      "Interleukin-20 family signaling: 7645.0\n",
      "Interleukin-1 signaling: 36678.0\n",
      "Interleukin-1 processing: 3284.0\n",
      "Interleukin-1 family signaling: 51766.0\n",
      "Interleukin-18 signaling: 6887.0\n",
      "Interleukin-17 signaling: 32917.0\n",
      "Interleukin-15 signaling: 3250.0\n",
      "Interleukin-12 signaling: 24155.0\n",
      "Interleukin-12 family signaling: 27951.0\n",
      "Interleukin-10 signaling: 26930.0\n",
      "Interferon Signaling: 69424.0\n",
      "Interferon gamma signaling: 26439.0\n",
      "Interferon alpha/beta signaling: 28627.0\n",
      "Interconversion of polyamines: 54.0\n",
      "Interconversion of nucleotide di- and triphosphates: 1507.0\n",
      "Interaction With Cumulus Cells And The Zona Pellucida: 2085.0\n",
      "Interaction between L1 and Ankyrins: 3589.0\n",
      "Integrin signaling: 8183.0\n",
      "Integrin cell surface interactions: 39811.0\n",
      "Integration of energy metabolism: 28870.0\n",
      "Insulin receptor signalling cascade: 23286.0\n",
      "Insulin receptor recycling: 9592.0\n",
      "Insulin processing: 8558.0\n",
      "Insulin-like Growth Factor-2 mRNA Binding Proteins (IGF2BPs/IMPs/VICKZs) bind RNA: 110.0\n",
      "Insertion of tail-anchored proteins into the endoplasmic reticulum membrane: 11351.0\n",
      "Inositol phosphate metabolism: 13450.0\n",
      "Innate Immune System: 378599.0\n",
      "Initiation of Nuclear Envelope Reformation: 4834.0\n",
      "Initial triggering of complement: 3478.0\n",
      "Inhibition of voltage gated Ca2+ channels via Gbeta/gamma subunits: 5090.0\n",
      "Inhibition of TSC complex formation by PKB: 129.0\n",
      "Inhibition of the proteolytic activity of APC/C required for the onset of anaphase by mitotic spindle checkpoint components: 5903.0\n",
      "Inhibition of replication initiation of damaged DNA by RB1/E2F1: 6967.0\n",
      "Inflammasomes: 4256.0\n",
      "Incretin synthesis: 9046.0\n",
      "Inactivation of CDC42 and RAC1: 154.0\n",
      "Inactivation of APC/C via direct inhibition of the APC/C complex: 5903.0\n",
      "Inactivation: 14758.0\n",
      "Immunoregulatory interactions between a Lymphoid and a non-Lymphoid cell: 56187.0\n",
      "IL-6-type cytokine receptor ligand interactions: 3793.0\n",
      "IKK complex recruitment mediated by RIP1: 3715.0\n",
      "IGF1R signaling cascade: 21982.0\n",
      "Hypusine synthesis from eIF5A-lysine: 184.0\n",
      "Hydroxycarboxylic acid-binding receptors: 0.0\n",
      "Hydrolysis of LPC: 5939.0\n",
      "Hyaluronan uptake and degradation: 2229.0\n",
      "Hyaluronan metabolism: 2457.0\n",
      "Hyaluronan biosynthesis and export: 228.0\n",
      "HuR (ELAVL1) binds and stabilizes mRNA: 2025.0\n",
      "HSP90 chaperone cycle for steroid hormone receptors (SHR): 8320.0\n",
      "HS-GAG degradation: 7509.0\n",
      "HS-GAG biosynthesis: 12467.0\n",
      "HSF1-dependent transactivation: 5820.0\n",
      "HSF1 activation: 4604.0\n",
      "Hormone ligand-binding receptors: 3158.0\n",
      "Homology Directed Repair: 31817.0\n",
      "Homologous DNA Pairing and Strand Exchange: 13368.0\n",
      "Histidine catabolism: 0.0\n",
      "Histamine receptors: 10554.0\n",
      "Highly sodium permeable acetylcholine nicotinic receptors: 9836.0\n",
      "Highly calcium permeable postsynaptic nicotinic acetylcholine receptors: 9719.0\n",
      "Highly calcium permeable nicotinic acetylcholine receptors: 9719.0\n",
      "Heparan sulfate/heparin (HS-GAG) metabolism: 17727.0\n",
      "Heme degradation: 3570.0\n",
      "Heme biosynthesis: 3754.0\n",
      "Hedgehog ligand biogenesis: 22458.0\n",
      "Hedgehog 'on' state: 31256.0\n",
      "Hedgehog 'off' state: 38471.0\n",
      "HDR through Single Strand Annealing (SSA): 10652.0\n",
      "HDR through MMEJ (alt-NHEJ): 4526.0\n",
      "HDR through Homologous Recombination (HRR) or Single Strand Annealing (SSA): 31253.0\n",
      "HDR through Homologous Recombination (HRR): 13368.0\n",
      "HDMs demethylate histones: 9998.0\n",
      "HDL remodeling: 5218.0\n",
      "HDL clearance: 4865.0\n",
      "HDL assembly: 1802.0\n",
      "HDACs deacetylate histones: 11859.0\n",
      "HCN channels: 136.0\n",
      "HATs acetylate histones: 27762.0\n",
      "GTP hydrolysis and joining of the 60S ribosomal subunit: 80211.0\n",
      "Growth hormone receptor signaling: 12577.0\n",
      "GRB7 events in ERBB2 signaling: 1574.0\n",
      "GRB2 events in ERBB2 signaling: 4432.0\n",
      "GRB2 events in EGFR signaling: 5953.0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GRB2:SOS provides linkage to MAPK signaling for Integrins: 7396.0\n",
      "GPVI-mediated activation cascade: 4976.0\n",
      "G-protein mediated events: 27199.0\n",
      "G protein gated Potassium channels: 5090.0\n",
      "G-protein beta:gamma signalling: 1664.0\n",
      "G-protein activation: 12879.0\n",
      "GPCR ligand binding: 158935.0\n",
      "GPCR downstream signalling: 255252.0\n",
      "GP1b-IX-V activation signalling: 1467.0\n",
      "Golgi-to-ER retrograde transport: 42311.0\n",
      "Golgi Cisternae Pericentriolar Stack Reorganization: 4965.0\n",
      "Golgi Associated Vesicle Biogenesis: 13912.0\n",
      "Glyoxylate metabolism and glycine degradation: 15451.0\n",
      "Glycosphingolipid metabolism: 8862.0\n",
      "Glycosaminoglycan metabolism: 45779.0\n",
      "Glycoprotein hormones: 3644.0\n",
      "Glycolysis: 31217.0\n",
      "Glycogen synthesis: 6739.0\n",
      "Glycogen metabolism: 8438.0\n",
      "Glycogen breakdown (glycogenolysis): 1761.0\n",
      "Glycine degradation: 313.0\n",
      "Glycerophospholipid biosynthesis: 38570.0\n",
      "Glutathione synthesis and recycling: 2377.0\n",
      "Glutathione conjugation: 3180.0\n",
      "Glutamate Neurotransmitter Release Cycle: 29695.0\n",
      "Glutamate binding: 5625.0\n",
      "Glutamate and glutamine metabolism: 3294.0\n",
      "Glucuronidation: 1306.0\n",
      "Glucose metabolism: 38359.0\n",
      "Gluconeogenesis: 17920.0\n",
      "Glucocorticoid biosynthesis: 3489.0\n",
      "Glucagon-type ligand receptors: 10080.0\n",
      "Glucagon signaling in metabolic regulation: 11103.0\n",
      "Glucagon-like Peptide-1 (GLP1) regulates insulin secretion: 11725.0\n",
      "Global Genome Nucleotide Excision Repair (GG-NER): 30870.0\n",
      "GLI proteins bind promoters of Hh responsive genes to promote transcription: 1983.0\n",
      "GLI3 is processed to GLI3R by the proteasome: 26887.0\n",
      "Gene Silencing by RNA: 28715.0\n",
      "Generic Transcription Pathway: 391061.0\n",
      "Generation of second messenger molecules: 7325.0\n",
      "Gene and protein expression by JAK-STAT signaling after Interleukin-12 stimulation: 16086.0\n",
      "G beta:gamma signalling through PLC beta: 164.0\n",
      "G beta:gamma signalling through PI3Kgamma: 1354.0\n",
      "G beta:gamma signalling through CDC42: 318.0\n",
      "G beta:gamma signalling through BTK: 320.0\n",
      "Gastrin-CREB signalling pathway via PKC and MAPK: 8892.0\n",
      "Gap junction trafficking and regulation: 32.0\n",
      "Gap junction trafficking: 0.0\n",
      "Gap junction degradation: 0.0\n",
      "Gap junction assembly: 0.0\n",
      "Gap-filling DNA repair synthesis and ligation in TC-NER: 17088.0\n",
      "Gap-filling DNA repair synthesis and ligation in GG-NER: 8498.0\n",
      "Gamma-carboxylation of protein precursors: 8236.0\n",
      "Gamma-carboxylation: 8236.0\n",
      "Gamma carboxylation: 15169.0\n",
      "G alpha (z) signalling events: 7195.0\n",
      "G alpha (s) signalling events: 53891.0\n",
      "G alpha (q) signalling events: 75214.0\n",
      "G alpha (i) signalling events: 146114.0\n",
      "G alpha (12/13) signalling events: 22627.0\n",
      "Galactose catabolism: 1583.0\n",
      "GABA synthesis: 23797.0\n",
      "GABA receptor activation: 13579.0\n",
      "GABA B receptor activation: 10328.0\n",
      "GAB1 signalosome: 7827.0\n",
      "G2 Phase: 6403.0\n",
      "G2/M Transition: 63600.0\n",
      "G2/M DNA replication checkpoint: 0.0\n",
      "G2/M DNA damage checkpoint: 18625.0\n",
      "G2/M Checkpoints: 51280.0\n",
      "G1 Phase: 16018.0\n",
      "G1/S Transition: 86542.0\n",
      "G1/S-Specific Transcription: 35887.0\n",
      "G1/S DNA Damage Checkpoints: 26149.0\n",
      "G0 and Early G1: 9725.0\n",
      "Fructose metabolism: 92.0\n",
      "Fructose catabolism: 0.0\n",
      "FRS-mediated FGFR4 signaling: 9206.0\n",
      "FRS-mediated FGFR3 signaling: 7839.0\n",
      "FRS-mediated FGFR2 signaling: 9913.0\n",
      "FRS-mediated FGFR1 signaling: 9771.0\n",
      "Frs2-mediated activation: 4608.0\n",
      "Free fatty acids regulate insulin secretion: 3726.0\n",
      "Free fatty acid receptors: 4617.0\n",
      "FOXO-mediated transcription of oxidative stress: 15586.0\n",
      "FOXO-mediated transcription of cell death genes: 28083.0\n",
      "FOXO-mediated transcription of cell cycle genes: 7172.0\n",
      "FOXO-mediated transcription: 49473.0\n",
      "Formyl peptide receptors bind formyl peptides and many other ligands: 134.0\n",
      "Formation of xylulose-5-phosphate: 209.0\n",
      "Formation of tubulin folding intermediates by CCT/TriC: 13048.0\n",
      "Formation of the ternary complex: 43941.0\n",
      "Formation of the Early Elongation Complex: 20263.0\n",
      "Formation of the cornified envelope: 34105.0\n",
      "Formation of the beta-catenin:TCF transactivating complex: 7975.0\n",
      "Formation of the active cofactor: 1306.0\n",
      "Formation of TC-NER Pre-Incision Complex: 12151.0\n",
      "Formation of Senescence-Associated Heterochromatin Foci (SAHF): 5062.0\n",
      "Formation of RNA Pol II elongation complex: 23827.0\n",
      "Formation of Incision Complex in GG-NER: 10322.0\n",
      "Formation of Fibrin Clot (Clotting Cascade): 17199.0\n",
      "Formation of ATP by chemiosmotic coupling: 0.0\n",
      "Formation of apoptosome: 7178.0\n",
      "Formation of a pool of free 40S subunits: 62077.0\n",
      "Folding of actin by CCT/TriC: 3182.0\n",
      "FLT3 Signaling: 122465.0\n",
      "Ficolins bind to repetitive carbohydrate structures on the target cell surface: 281.0\n",
      "Fibronectin matrix formation: 3848.0\n",
      "FGFRL1 modulation of FGFR1 signaling: 2046.0\n",
      "FGFR4 ligand binding and activation: 7624.0\n",
      "FGFR3 ligand binding and activation: 8202.0\n",
      "FGFR3c ligand binding and activation: 8202.0\n",
      "FGFR3b ligand binding and activation: 4550.0\n",
      "FGFR2 ligand binding and activation: 8331.0\n",
      "FGFR2c ligand binding and activation: 7954.0\n",
      "FGFR2b ligand binding and activation: 2293.0\n",
      "FGFR2 alternative splicing: 9340.0\n",
      "FGFR1 ligand binding and activation: 8189.0\n",
      "FGFR1c ligand binding and activation: 7812.0\n",
      "FGFR1c and Klotho ligand binding and activation: 1795.0\n",
      "FGFR1b ligand binding and activation: 2151.0\n",
      "Fertilization: 2454.0\n",
      "FCGR activation: 6741.0\n",
      "Fcgamma receptor (FCGR) dependent phagocytosis: 26066.0\n",
      "FCERI mediated NF-kB activation: 25788.0\n",
      "FCERI mediated MAPK activation: 8494.0\n",
      "FCERI mediated Ca+2 mobilization: 9000.0\n",
      "Fc epsilon receptor (FCERI) signaling: 43273.0\n",
      "FBXL7 down-regulates AURKA during mitotic entry and in early mitosis: 23362.0\n",
      "Fatty acyl-CoA biosynthesis: 6122.0\n",
      "Fatty acids: 0.0\n",
      "Fatty acid metabolism: 49549.0\n",
      "FasL/ CD95L signaling: 5932.0\n",
      "Fanconi Anemia Pathway: 6770.0\n",
      "Factors involved in megakaryocyte development and platelet production: 41181.0\n",
      "Extrinsic Pathway of Fibrin Clot Formation: 5465.0\n",
      "Extra-nuclear estrogen signaling: 24804.0\n",
      "Extension of Telomeres: 10344.0\n",
      "Expression and Processing of Neurotrophins: 155.0\n",
      "Eukaryotic Translation Termination: 61675.0\n",
      "Eukaryotic Translation Initiation: 84855.0\n",
      "Eukaryotic Translation Elongation: 62027.0\n",
      "Ethanol oxidation: 1498.0\n",
      "Estrogen-stimulated signaling through PRKCZ: 391.0\n",
      "Estrogen-dependent nuclear events downstream of ESR-membrane signaling: 14824.0\n",
      "Estrogen-dependent gene expression: 48358.0\n",
      "Estrogen biosynthesis: 1705.0\n",
      "Establishment of Sister Chromatid Cohesion: 6661.0\n",
      "ESR-mediated signaling: 69596.0\n",
      "Erythropoietin activates STAT5: 1556.0\n",
      "Erythropoietin activates RAS: 3423.0\n",
      "Erythropoietin activates Phospholipase C gamma (PLCG): 2027.0\n",
      "Erythropoietin activates Phosphoinositide-3-kinase (PI3K): 4425.0\n",
      "Erythrocytes take up oxygen and release carbon dioxide: 1995.0\n",
      "Erythrocytes take up carbon dioxide and release oxygen: 2405.0\n",
      "ER to Golgi Anterograde Transport: 105261.0\n",
      "ER Quality Control Compartment (ERQC): 1303.0\n",
      "ER-Phagosome pathway: 26140.0\n",
      "ERKs are inactivated: 5581.0\n",
      "ERK/MAPK targets: 10685.0\n",
      "ERCC6 (CSB) and EHMT2 (G9a) positively regulate rRNA expression: 8026.0\n",
      "ERBB2 Regulates Cell Motility: 4713.0\n",
      "ERBB2 Activates PTK6 Signaling: 11159.0\n",
      "Epigenetic regulation of gene expression: 33749.0\n",
      "Ephrin signaling: 8765.0\n",
      "EPH-Ephrin signaling: 38385.0\n",
      "EPH-ephrin mediated repulsion of cells: 10000.0\n",
      "EPHB-mediated forward signaling: 4955.0\n",
      "EPHA-mediated growth cone collapse: 19204.0\n",
      "Enzymatic degradation of dopamine by COMT: 208.0\n",
      "eNOS activation: 4399.0\n",
      "Energy dependent regulation of mTOR by LKB1-AMPK: 11983.0\n",
      "Endosomal Sorting Complex Required For Transport (ESCRT): 8696.0\n",
      "Endosomal/Vacuolar pathway: 1589.0\n",
      "Endogenous sterols: 8273.0\n",
      "EML4 and NUDC in mitotic spindle formation: 32536.0\n",
      "Elevation of cytosolic Ca2+ levels: 29709.0\n",
      "Electron transport from NADPH to Ferredoxin: 0.0\n",
      "Electric Transmission Across Gap Junctions: 1766.0\n",
      "Elastic fibre formation: 31345.0\n",
      "Eicosanoids: 1829.0\n",
      "Eicosanoid ligand-binding receptors: 3322.0\n",
      "EGFR Transactivation by Gastrin: 3107.0\n",
      "EGFR interacts with phospholipase C-gamma: 6239.0\n",
      "EGFR downregulation: 15153.0\n",
      "Effects of PIP2 hydrolysis: 5957.0\n",
      "ECM proteoglycans: 37753.0\n",
      "E3 ubiquitin ligases ubiquitinate target proteins: 5412.0\n",
      "E2F mediated regulation of DNA replication: 11793.0\n",
      "E2F-enabled inhibition of pre-replication complex formation: 4826.0\n",
      "Dual incision in TC-NER: 13894.0\n",
      "Dual Incision in GG-NER: 8212.0\n",
      "DSCAM interactions: 4633.0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Drug-mediated inhibition of ERBB2 signaling: 3929.0\n",
      "Downstream TCR signaling: 29148.0\n",
      "Downstream signal transduction: 13497.0\n",
      "Downstream signaling of activated FGFR4: 14372.0\n",
      "Downstream signaling of activated FGFR3: 13005.0\n",
      "Downstream signaling of activated FGFR2: 15079.0\n",
      "Downstream signaling of activated FGFR1: 18046.0\n",
      "Downstream signaling events of B Cell Receptor (BCR): 26830.0\n",
      "Downregulation of TGF-beta receptor signaling: 26820.0\n",
      "Downregulation of SMAD2/3:SMAD4 transcriptional activity: 8204.0\n",
      "Downregulation of ERBB4 signaling: 2566.0\n",
      "Downregulation of ERBB2 signaling: 8754.0\n",
      "Downregulation of ERBB2:ERBB3 signaling: 1574.0\n",
      "Dopamine receptors: 934.0\n",
      "Dopamine Neurotransmitter Release Cycle: 5387.0\n",
      "Dopamine clearance from the synaptic cleft: 208.0\n",
      "DNA strand elongation: 12310.0\n",
      "DNA Replication Pre-Initiation: 44623.0\n",
      "DNA replication initiation: 3971.0\n",
      "DNA methylation: 1375.0\n",
      "DNA Double Strand Break Response: 15245.0\n",
      "DNA Double-Strand Break Repair: 38269.0\n",
      "DNA Damage Reversal: 1927.0\n",
      "DNA Damage Recognition in GG-NER: 15265.0\n",
      "DNA Damage Bypass: 14614.0\n",
      "DNA Damage/Telomere Stress Induced Senescence: 11830.0\n",
      "Dissolution of Fibrin Clot: 5569.0\n",
      "Displacement of DNA glycosylase by APEX1: 5628.0\n",
      "Disinhibition of SNARE formation: 3143.0\n",
      "Disassembly of the destruction complex and recruitment of AXIN to the membrane: 6585.0\n",
      "Dimerization of procaspase-8: 10149.0\n",
      "Digestion of dietary lipid: 0.0\n",
      "Digestion of dietary carbohydrate: 3457.0\n",
      "Digestion: 3820.0\n",
      "DEx/H-box helicases activate type I IFN and inflammatory cytokines production: 1698.0\n",
      "Deubiquitination: 109812.0\n",
      "Detoxification of Reactive Oxygen Species: 13675.0\n",
      "Dermatan sulfate biosynthesis: 5152.0\n",
      "Depyrimidination: 7866.0\n",
      "Depurination: 4295.0\n",
      "Deposition of new CENPA-containing nucleosomes at the centromere: 10810.0\n",
      "Depolymerisation of the Nuclear Lamina: 1387.0\n",
      "Degradation of the extracellular matrix: 20348.0\n",
      "Degradation of GLI2 by the proteasome: 26828.0\n",
      "Degradation of GLI1 by the proteasome: 26913.0\n",
      "Degradation of DVL: 26209.0\n",
      "Degradation of cysteine and homocysteine: 4603.0\n",
      "Degradation of beta-catenin by the destruction complex: 28991.0\n",
      "Degradation of AXIN: 25986.0\n",
      "Defensins: 3746.0\n",
      "Dectin-2 family: 1837.0\n",
      "Dectin-1 mediated noncanonical NF-kB signaling: 23717.0\n",
      "Death Receptor Signalling: 81763.0\n",
      "Deadenylation of mRNA: 21412.0\n",
      "Deadenylation-dependent mRNA decay: 34598.0\n",
      "Deactivation of the beta-catenin transactivating complex: 13873.0\n",
      "DDX58/IFIH1-mediated induction of interferon-alpha/beta: 24469.0\n",
      "DCC mediated attractive signaling: 2451.0\n",
      "DARPP-32 events: 11915.0\n",
      "DAP12 signaling: 7452.0\n",
      "DAP12 interactions: 13108.0\n",
      "DAG and IP3 signaling: 18101.0\n",
      "Cytosolic tRNA aminoacylation: 3247.0\n",
      "Cytosolic sulfonation of small molecules: 8363.0\n",
      "Cytosolic sensors of pathogen-associated DNA: 13655.0\n",
      "Cytosolic iron-sulfur cluster assembly: 604.0\n",
      "Cytokine Signaling in Immune system: 364250.0\n",
      "Cytochrome P450 - arranged by substrate type: 24066.0\n",
      "Cytochrome c-mediated apoptotic response: 7178.0\n",
      "Cyclin E associated events during G1/S transition: 66739.0\n",
      "Cyclin D associated events in G1: 16018.0\n",
      "Cyclin A/B1/B2 associated events during G2/M transition: 8662.0\n",
      "Cyclin A:Cdk2-associated events at S phase entry: 82826.0\n",
      "C-type lectin receptors (CLRs): 39022.0\n",
      "CTLA4 inhibitory signaling: 3720.0\n",
      "CS/DS degradation: 5133.0\n",
      "Cross-presentation of soluble exogenous antigens (endosomes): 21732.0\n",
      "Cross-presentation of particulate exogenous antigens (phagosomes): 1951.0\n",
      "Crosslinking of collagen fibrils: 1591.0\n",
      "CRMPs in Sema3A signaling: 4166.0\n",
      "Cristae formation: 2075.0\n",
      "CREB phosphorylation: 4831.0\n",
      "CREB3 factors activate genes: 1859.0\n",
      "CREB1 phosphorylation through the activation of CaMKII/CaMKK/CaMKIV cascasde: 3608.0\n",
      "CREB1 phosphorylation through the activation of Adenylate Cyclase: 5289.0\n",
      "CREB1 phosphorylation through NMDA receptor-mediated activation of RAS signaling: 6872.0\n",
      "Creation of C4 and C2 activators: 1873.0\n",
      "Creatine metabolism: 20.0\n",
      "Costimulation by the CD28 family: 17326.0\n",
      "COPI-mediated anterograde transport: 37296.0\n",
      "COPI-independent Golgi-to-ER retrograde traffic: 8145.0\n",
      "COPII-mediated vesicle transport: 77565.0\n",
      "COPI-dependent Golgi-to-ER retrograde traffic: 34166.0\n",
      "Cooperation of Prefoldin and TriC/CCT in actin and tubulin folding: 16244.0\n",
      "Cooperation of PDCL (PhLP1) and TRiC/CCT in G-protein beta folding: 18676.0\n",
      "Conversion from APC/C:Cdc20 to APC/C:Cdh1 in late anaphase: 21990.0\n",
      "Conjugation of carboxylic acids: 0.0\n",
      "Conjugation of benzoate with glycine: 0.0\n",
      "Condensation of Prophase Chromosomes: 2985.0\n",
      "Condensation of Prometaphase Chromosomes: 9241.0\n",
      "Complex I biogenesis: 20629.0\n",
      "Complement cascade: 24247.0\n",
      "Competing endogenous RNAs (ceRNAs) regulate PTEN translation: 2581.0\n",
      "Common Pathway of Fibrin Clot Formation: 8567.0\n",
      "Collagen formation: 22343.0\n",
      "Collagen degradation: 10511.0\n",
      "Collagen chain trimerization: 17821.0\n",
      "Collagen biosynthesis and modifying enzymes: 20218.0\n",
      "Cohesin Loading onto Chromatin: 6661.0\n",
      "Cobalamin (Cbl: 6581.0\n",
      "CLEC7A (Dectin-1) signaling: 34938.0\n",
      "CLEC7A (Dectin-1) induces NFAT activation: 3754.0\n",
      "CLEC7A/inflammasome pathway: 7142.0\n",
      "Cleavage of the damaged pyrimidine: 7866.0\n",
      "Cleavage of the damaged purine: 4295.0\n",
      "Clathrin-mediated endocytosis: 59966.0\n",
      "Class I peroxisomal membrane protein import: 11226.0\n",
      "Class I MHC mediated antigen processing & presentation: 63675.0\n",
      "Classical Kir channels: 1583.0\n",
      "Classical antibody-mediated complement activation: 0.0\n",
      "Class C/3 (Metabotropic glutamate/pheromone receptors): 9563.0\n",
      "Class B/2 (Secretin family receptors): 25194.0\n",
      "Class A/1 (Rhodopsin-like receptors): 124178.0\n",
      "Citric acid cycle (TCA cycle): 8711.0\n",
      "Cilium Assembly: 75382.0\n",
      "Chylomicron remodeling: 5831.0\n",
      "Chylomicron clearance: 117.0\n",
      "Chylomicron assembly: 4267.0\n",
      "Chromosome Maintenance: 21940.0\n",
      "Chromatin modifying enzymes: 67914.0\n",
      "ChREBP activates metabolic gene expression: 1838.0\n",
      "Chondroitin sulfate biosynthesis: 5380.0\n",
      "Chondroitin sulfate/dermatan sulfate metabolism: 22653.0\n",
      "Choline catabolism: 202.0\n",
      "Cholesterol biosynthesis via lathosterol: 3568.0\n",
      "Cholesterol biosynthesis via desmosterol: 3568.0\n",
      "Cholesterol biosynthesis: 11659.0\n",
      "CHL1 interactions: 1915.0\n",
      "Chk1/Chk2(Cds1) mediated inactivation of Cyclin B:Cdk1 complex: 1336.0\n",
      "Chemokine receptors bind chemokines: 19490.0\n",
      "Chaperonin-mediated protein folding: 56813.0\n",
      "Chaperone Mediated Autophagy: 18786.0\n",
      "cGMP effects: 5120.0\n",
      "Ceramide signalling: 25836.0\n",
      "Centrosome maturation: 26672.0\n",
      "Cellular Senescence: 67171.0\n",
      "Cellular response to hypoxia: 31827.0\n",
      "Cellular response to heat stress: 29454.0\n",
      "Cellular responses to stress: 233401.0\n",
      "Cellular hexose transport: 5194.0\n",
      "Cell surface interactions at the vascular wall: 74839.0\n",
      "Cell junction organization: 11358.0\n",
      "Cell-extracellular matrix interactions: 3504.0\n",
      "Cell death signalling via NRAGE: 52180.0\n",
      "Cell Cycle Checkpoints: 88363.0\n",
      "Cell Cycle: 216150.0\n",
      "Cell-cell junction organization: 7007.0\n",
      "CDT1 association with the CDC6:ORC:origin complex: 28414.0\n",
      "CDK-mediated phosphorylation and removal of Cdc6: 68876.0\n",
      "CDC6 association with the ORC:origin complex: 10194.0\n",
      "Cdc20:Phospho-APC/C mediated degradation of Cyclin A: 27635.0\n",
      "CD28 dependent Vav1 pathway: 3878.0\n",
      "CD28 dependent PI3K/Akt signaling: 13663.0\n",
      "CD28 co-stimulation: 13872.0\n",
      "CD22 mediated BCR regulation: 13562.0\n",
      "CD209 (DC-SIGN) signaling: 4079.0\n",
      "Cation-coupled Chloride cotransporters: 6333.0\n",
      "Catecholamine biosynthesis: 1819.0\n",
      "Caspase-mediated cleavage of cytoskeletal proteins: 4019.0\n",
      "Caspase activation via extrinsic apoptotic signalling pathway: 13054.0\n",
      "Caspase activation via Dependence Receptors in the absence of ligand: 2905.0\n",
      "Caspase activation via Death Receptors in the presence of ligand: 10149.0\n",
      "CASP8 activity is inhibited: 10149.0\n",
      "Carnitine metabolism: 3385.0\n",
      "Cargo trafficking to the periciliary membrane: 19527.0\n",
      "Cargo recognition for clathrin-mediated endocytosis: 40817.0\n",
      "Cargo concentration in the ER: 20904.0\n",
      "Cardiac conduction: 48273.0\n",
      "Carboxyterminal post-translational modifications of tubulin: 1637.0\n",
      "Cap-dependent Translation Initiation: 84855.0\n",
      "Cam-PDE 1 activation: 3636.0\n",
      "CaM pathway: 14133.0\n",
      "CaMK IV-mediated phosphorylation of CREB: 3766.0\n",
      "Calnexin/calreticulin cycle: 1353.0\n",
      "Calmodulin induced events: 14133.0\n",
      "Calcitonin-like ligand receptors: 7143.0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calcineurin activates NFAT: 1708.0\n",
      "Ca-dependent events: 16053.0\n",
      "Ca2+ pathway: 8477.0\n",
      "Ca2+ activated K+ channels: 1701.0\n",
      "B-WICH complex positively regulates rRNA expression: 12646.0\n",
      "Butyrophilin (BTN) family interactions: 7859.0\n",
      "Butyrate Response Factor 1 (BRF1) binds and destabilizes mRNA: 8307.0\n",
      "Breakdown of the nuclear lamina: 1583.0\n",
      "Branched-chain amino acid catabolism: 3527.0\n",
      "BMAL1:CLOCK: 16846.0\n",
      "Blood group systems biosynthesis: 8998.0\n",
      "Biotin transport and metabolism: 3727.0\n",
      "Biosynthesis of the N-glycan precursor (dolichol lipid-linked oligosaccharide: 27965.0\n",
      "Biosynthesis of specialized proresolving mediators (SPMs): 5281.0\n",
      "Biosynthesis of protectins: 1526.0\n",
      "Biosynthesis of maresins: 2104.0\n",
      "Biosynthesis of maresin-like SPMs: 98.0\n",
      "Biosynthesis of E-series 18(S)-resolvins: 1671.0\n",
      "Biosynthesis of E-series 18(R)-resolvins: 1671.0\n",
      "Biosynthesis of EPA-derived SPMs: 1778.0\n",
      "Biosynthesis of D-series resolvins: 1671.0\n",
      "Biosynthesis of DPAn-3 SPMs: 1796.0\n",
      "Biosynthesis of DPA-derived SPMs: 1796.0\n",
      "Biosynthesis of DHA-derived SPMs: 5281.0\n",
      "Biosynthesis of aspirin-triggered D-series resolvins: 1671.0\n",
      "Biological oxidations: 54490.0\n",
      "Binding of TCF/LEF:CTNNB1 to target gene promoters: 235.0\n",
      "Binding and Uptake of Ligands by Scavenger Receptors: 13065.0\n",
      "Bile acid and bile salt metabolism: 14505.0\n",
      "Bicarbonate transporters: 640.0\n",
      "BH3-only proteins associate with and inactivate anti-apoptotic BCL-2 members: 19057.0\n",
      "Beta-oxidation of very long chain fatty acids: 5704.0\n",
      "Beta-oxidation of pristanoyl-CoA: 4982.0\n",
      "Beta oxidation of palmitoyl-CoA to myristoyl-CoA: 110.0\n",
      "Beta oxidation of octanoyl-CoA to hexanoyl-CoA: 2007.0\n",
      "Beta oxidation of myristoyl-CoA to lauroyl-CoA: 1739.0\n",
      "Beta oxidation of lauroyl-CoA to decanoyl-CoA-CoA: 1739.0\n",
      "Beta oxidation of hexanoyl-CoA to butanoyl-CoA: 110.0\n",
      "Beta oxidation of decanoyl-CoA to octanoyl-CoA-CoA: 2007.0\n",
      "Beta oxidation of butanoyl-CoA to acetyl-CoA: 395.0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "IOPub message rate exceeded.\n",
      "The notebook server will temporarily stop sending output\n",
      "to the client in order to avoid crashing it.\n",
      "To change this limit, set the config variable\n",
      "`--NotebookApp.iopub_msg_rate_limit`.\n",
      "\n",
      "Current values:\n",
      "NotebookApp.iopub_msg_rate_limit=1000.0 (msgs/sec)\n",
      "NotebookApp.rate_limit_window=3.0 (secs)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for column in pathway_encodings.columns:\n",
    "    if column != \"CancerType\":\n",
    "        print(\"{}: {}\".format(column, sum(pathway_encodings[column])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "pathway_encodings.to_csv(\"pathway_encodings_full.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
